Index: web-plugins-0.2.0/web-plugins.cabal
===================================================================
--- web-plugins-0.2.0.orig/web-plugins.cabal	2013-01-09 19:10:58.000000000 +0000
+++ web-plugins-0.2.0/web-plugins.cabal	2013-02-09 22:13:49.381968527 +0000
@@ -21,4 +21,4 @@
                        stm        == 2.4.*,
                        mtl        == 2.1.*,
                        text       == 0.11.*,
-                       containers == 0.4.*
+                       containers == 0.5.*
