--- a/Hledger/Web/Application.hs
+++ b/Hledger/Web/Application.hs
@@ -15,7 +15,7 @@
 import Yesod.Logger (makeLogger, flushLogger, Logger, logLazyText, logString)
 import Yesod.Static

-import Hledger.Web.Foundation
+import Hledger.Web.Foundation hiding (AppConfig,loadConfig,appPort)
 import Hledger.Web.Handlers
 import Hledger.Web.Options
 import Hledger.Web.Settings
--- a/Hledger/Web.hs
+++ b/Hledger/Web.hs
@@ -14,7 +14,7 @@
 where
 import Test.HUnit

-import Hledger.Web.Foundation
+import Hledger.Web.Foundation hiding (AppConfig, loadConfig, appPort, appEnv, appRoot)
 import Hledger.Web.Application
 import Hledger.Web.Handlers
 import Hledger.Web.Options
