--- a/Hledger/Web/Settings/StaticFiles.hs
+++ b/Hledger/Web/Settings/StaticFiles.hs
@@ -11,19 +11,22 @@
 -}
 module Hledger.Web.Settings.StaticFiles where

-import Prelude (IO)
+import System.IO
 import Yesod.Static
 import qualified Yesod.Static as Static

+import Prelude
 import Hledger.Web.Settings (staticDir)

 -- | use this to create your static file serving site
 staticSite :: IO Static.Static
-staticSite =
+staticSite = do
 #ifdef DEVELOPMENT
+  putStrLn ("using web files from: " ++ staticDir ++ "/") >> hFlush stdout
   Static.staticDevel staticDir
 #else
-  Static.static staticDir
+  putStrLn "using embedded web files" >> hFlush stdout
+  return $(Static.embed staticDir)
 #endif

 -- | This generates easy references to files in the static directory at compile time,
