--- a/hledger-web.cabal
+++ b/hledger-web.cabal
@@ -149,8 +149,8 @@ library
                    , transformers
                    , wai
                    , wai-extra
-                   , wai-handler-launch   >= 1.3 && < 2.1
-                   , warp                 >= 1.3 && < 2.2
+                   , wai-handler-launch   >= 3.0 && < 3.1
+                   , warp                 >= 3.0 && < 3.1
                    , yaml
                    , yesod                >= 1.2 && < 1.3
                    , yesod-core
@@ -240,8 +240,8 @@ executable         hledger-web
                    , transformers
                    , wai
                    , wai-extra
-                   , wai-handler-launch   >= 1.3 && < 2.1
-                   , warp                 >= 1.3 && < 2.2
+                   , wai-handler-launch   >= 3.0 && < 3.1
+                   , warp                 >= 3.0 && < 3.1
                    , yaml
                    , yesod                >= 1.2 && < 1.3
                    , yesod-core
