--- a/hledger-web.cabal
+++ b/hledger-web.cabal
@@ -140,8 +140,8 @@
                    , transformers
                    , wai
                    , wai-extra
-                   , wai-handler-launch   >= 1.3 && < 2.1
-                   , warp                 >= 1.3 && < 2.2
+                   , wai-handler-launch   >= 3.0 && < 3.1
+                   , warp                 >= 3.0 && < 3.1
                    , yaml
                    , yesod                >= 1.2 && < 1.3
                    , yesod-core
@@ -232,8 +232,8 @@
                    , transformers
                    , wai
                    , wai-extra
-                   , wai-handler-launch   >= 1.3 && < 2.1
-                   , warp                 >= 1.3 && < 2.2
+                   , wai-handler-launch   >= 3.0 && < 3.1
+                   , warp                 >= 3.0 && < 3.1
                    , yaml
                    , yesod                >= 1.2 && < 1.3
                    , yesod-core
