--- a/hledger-web.cabal
+++ b/hledger-web.cabal
@@ -127,7 +127,6 @@ library
                    , http-conduit
                    , http-client          >= 0.2 && < 0.4
                    , HUnit
-                   , network-conduit
                    , conduit-extra
                    , old-locale
                    , parsec
@@ -227,7 +226,6 @@ executable         hledger-web
                    , http-conduit
                    , http-client          >= 0.2 && < 0.4
                    , HUnit
-                   , network-conduit
                    , conduit-extra
                    , old-locale
                    , parsec
