Index: hledger-web-0.24.1/hledger-web.cabal
===================================================================
--- hledger-web-0.24.1.orig/hledger-web.cabal	2015-06-23 12:35:51.293411272 +0200
+++ hledger-web-0.24.1/hledger-web.cabal	2015-06-23 12:35:51.289411182 +0200
@@ -171,7 +171,6 @@
                    , http-conduit
                    , http-client
                    , HUnit
-                   , network-conduit
                    , conduit-extra
                    , old-locale
                    , parsec               >= 3
@@ -242,7 +241,6 @@
                    , http-conduit
                    , http-client
                    , HUnit
-                   , network-conduit
                    , conduit-extra
                    , old-locale
                    , parsec               >= 3
