From 531b14ec3ec33d9b9f8535753bbd6b646d956eb6 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@devalot.com>
Date: Thu, 23 Mar 2017 16:58:35 -0700
Subject: [PATCH] Bump upper bounds on the hspec package

---
 map-syntax.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/map-syntax.cabal b/map-syntax.cabal
index 631036e..6feeed4 100644
--- a/map-syntax.cabal
+++ b/map-syntax.cabal
@@ -63,5 +63,5 @@ Test-suite testsuite
     HUnit                      >= 1.2      && < 2,
     mtl                        >= 2.0      && < 2.3,
     QuickCheck                 >= 2.3.0.2  && < 3,
-    hspec                      >= 2.2.3    && < 2.4,
+    hspec                      >= 2.2.3    && < 2.5,
     transformers               >= 0.3      && < 0.6
