From: Vincent Hanquez <vincent@snarc.org>
Date: Sat, 19 Jan 2013 11:09:46 +0000
Subject: [PATCH] properly implement basic constraints
Origin: upstream, https://github.com/vincenthz/hs-certificate/commit/a156d857189fc880f7d0a2de3310e750994c766b
Bug-Debian: http://bugs.debian.org/700284


diff -ur orig/haskell-certificate-1.2.3/Data/Certificate/X509/Ext.hs haskell-certificate-1.2.3/Data/Certificate/X509/Ext.hs
--- orig/haskell-certificate-1.2.3/Data/Certificate/X509/Ext.hs	2012-05-16 04:30:24.000000000 -0400
+++ haskell-certificate-1.2.3/Data/Certificate/X509/Ext.hs	2013-03-10 13:58:39.000000000 -0400
@@ -64,14 +64,19 @@
		| otherwise       -> extensionGet xs
	Left _                    -> extensionGet xs

-data ExtBasicConstraints = ExtBasicConstraints Bool
+data ExtBasicConstraints = ExtBasicConstraints Bool (Maybe Integer)
	deriving (Show,Eq)

 instance Extension ExtBasicConstraints where
	extOID = const [2,5,29,19]
-	extEncode (ExtBasicConstraints b) = [Start Sequence,Boolean b,End Sequence]
-	extDecode [Start Sequence,Boolean b,End Sequence] = Right (ExtBasicConstraints b)
-	extDecode [Start Sequence,End Sequence] = Right (ExtBasicConstraints False)
+	extEncode (ExtBasicConstraints b Nothing)  = [Start Sequence,Boolean b,End Sequence]
+	extEncode (ExtBasicConstraints b (Just i)) = [Start Sequence,Boolean b,IntVal i,End Sequence]
+
+	extDecode [Start Sequence,Boolean b,IntVal v,End Sequence]
+		| v >= 0    = Right (ExtBasicConstraints b (Just v))
+		| otherwise = Left "invalid pathlen"
+	extDecode [Start Sequence,Boolean b,End Sequence] = Right (ExtBasicConstraints b Nothing)
+	extDecode [Start Sequence,End Sequence] = Right (ExtBasicConstraints False Nothing)
	extDecode _ = Left "unknown sequence"

 data ExtKeyUsage = ExtKeyUsage [ExtKeyUsageFlag]
