Index: certificate-1.3.7/certificate.cabal
===================================================================
--- certificate-1.3.7.orig/certificate.cabal	2013-04-28 06:35:39.000000000 +0000
+++ certificate-1.3.7/certificate.cabal	2013-05-25 14:43:22.384370493 +0000
@@ -31,7 +31,7 @@
                    , mtl
                    , pem >= 0.1 && < 0.2
                    , asn1-data >= 0.7.1 && < 0.8.0
-                   , crypto-pubkey-types >= 0.3 && < 0.4
+                   , crypto-pubkey-types >= 0.4 && < 0.5
                    , cryptohash
                    , containers
                    , directory
