--- a/Plugin/Pl/Common.hs
+++ b/Plugin/Pl/Common.hs
@@ -1,5 +1,3 @@
-{-# OPTIONS -fvia-C #-}
-
 module Plugin.Pl.Common (
         Fixity(..), Expr(..), Pattern(..), Decl(..), TopLevel(..),
         bt, sizeExpr, mapTopLevel, getExpr,
--- a/Plugin/Pl/RuleLib.hs
+++ b/Plugin/Pl/RuleLib.hs
@@ -1,4 +1,4 @@
-{-# OPTIONS_GHC -fglasgow-exts #-} -- fix this later
+{-# OPTIONS_GHC -XScopedTypeVariables #-} -- fix this later
 {-# LANGUAGE FlexibleInstances, PatternGuards #-}

 -- | This marvellous module contributed by Thomas J\344ger
