Index: haskell-lens-4.1.2/lens.cabal
===================================================================
--- haskell-lens-4.1.2.orig/lens.cabal
+++ haskell-lens-4.1.2/lens.cabal
@@ -121,6 +121,12 @@ flag benchmark-uniplate
   default: False
   manual: True

+-- Avoid bulding modules that need TH
+flag without-th
+  default: False
+  manual: True
+
+
 -- Generate inline pragmas when using template-haskell. This defaults to enabled, but you can
 --
 -- > cabal install lens -f-inlining
@@ -248,7 +254,6 @@ library
     Control.Lens.Internal.Reflection
     Control.Lens.Internal.Review
     Control.Lens.Internal.Setter
-    Control.Lens.Internal.TH
     Control.Lens.Internal.Zoom
     Control.Lens.Iso
     Control.Lens.Lens
@@ -260,7 +265,6 @@ library
     Control.Lens.Reified
     Control.Lens.Review
     Control.Lens.Setter
-    Control.Lens.TH
     Control.Lens.Traversal
     Control.Lens.Tuple
     Control.Lens.Type
@@ -301,11 +305,18 @@ library
     Language.Haskell.TH.Lens
     Numeric.Lens

-  other-modules:
-    Control.Lens.Internal.TupleIxedTH

   cpp-options: -traditional

+  if flag(without-th)
+    cpp-options: -DDISABLE_TEMPLATE_HASKELL
+  else:
+    exposed-modules:
+      Control.Lens.TH
+      Control.Lens.Internal.TH
+    other-modules:
+      Control.Lens.Internal.TupleIxedTH
+
   if flag(safe)
     cpp-options: -DSAFE=1

@@ -339,6 +350,9 @@ test-suite templates
   ghc-options: -Wall -threaded
   hs-source-dirs: tests

+  if flag(without-th)
+    buildable: False
+
   if flag(dump-splices)
     ghc-options: -ddump-splices

Index: haskell-lens-4.1.2/src/Control/Lens/At.hs
===================================================================
--- haskell-lens-4.1.2.orig/src/Control/Lens/At.hs
+++ haskell-lens-4.1.2/src/Control/Lens/At.hs
@@ -2,7 +2,9 @@
 {-# LANGUAGE GADTs #-}
 {-# LANGUAGE Rank2Types #-}
 {-# LANGUAGE TypeFamilies #-}
+#ifndef DISABLE_TEMPLATE_HASKELL
 {-# LANGUAGE TemplateHaskell #-}
+#endif
 {-# LANGUAGE FlexibleContexts #-}
 {-# LANGUAGE DefaultSignatures #-}
 {-# LANGUAGE FlexibleInstances #-}
@@ -45,7 +47,9 @@ import Control.Applicative
 import Control.Lens.Lens
 import Control.Lens.Setter
 import Control.Lens.Type
+#ifndef DISABLE_TEMPLATE_HASKELL
 import Control.Lens.Internal.TupleIxedTH (makeAllTupleIxed)
+#endif
 import Data.Aeson as Aeson
 import Data.Array.IArray as Array
 import Data.Array.Unboxed
@@ -441,4 +445,6 @@ instance (Eq k, Hashable k) => At (HashS
     where mv = if HashSet.member k m then Just () else Nothing
   {-# INLINE at #-}

+#ifndef DISABLE_TEMPLATE_HASKELL
 makeAllTupleIxed
+#endif
Index: haskell-lens-4.1.2/src/System/IO/Error/Lens.hs
===================================================================
--- haskell-lens-4.1.2.orig/src/System/IO/Error/Lens.hs
+++ haskell-lens-4.1.2/src/System/IO/Error/Lens.hs
@@ -1,3 +1,4 @@
+{-# LANGUAGE CPP #-}
 {-# LANGUAGE TemplateHaskell #-}
 {-# LANGUAGE FlexibleContexts #-}
 {-# LANGUAGE FlexibleInstances #-}
@@ -62,4 +63,6 @@ errorType f s = f (ioe_type s) <&> \e ->
 --
 -- (These prisms are generated automatically)

+#ifndef DISABLE_TEMPLATE_HASKELL
 makePrisms ''IOErrorType
+#endif
