Description: Fix doctest with different float implementations
Author: Johan Kiviniemi <devel@johan.kiviniemi.name>
Origin: upstream, https://github.com/ekmett/lens/commit/ab2c8ee732ef09f713730ee8f431ef2f53535c3f
Bug: https://github.com/ekmett/lens/issues/310
Forwarded: not-needed
Last-Update: 2013-06-17

--- a/src/Control/Lens/Setter.hs
+++ b/src/Control/Lens/Setter.hs
@@ -671,8 +671,8 @@
 -- >>> (a,b) & both **~ c
 -- (a**c,b**c)
 --
--- >>> _2 **~ pi $ (1,3)
--- (1,31.54428070019754)
+-- >>> _2 **~ 10 $ (3,2)
+-- (3,1024.0)
 --
 -- @
 -- ('**~') :: 'Floating' a => 'Setter'' s a    -> a -> s -> s
