Index: b/src/Data/Complex/Lens.hs
===================================================================
--- a/src/Data/Complex/Lens.hs
+++ b/src/Data/Complex/Lens.hs
@@ -43,7 +43,7 @@ import Data.Complex
 -- >>> import Control.Lens
 -- >>> import Data.Complex
 -- >>> import Debug.SimpleReflect
--- >>> let { a ≈ b = abs (a - b) < 1e-6; infix 4 ≈ }
+-- >>> let { a ~~ b = abs (a - b) < 1e-6; infix 4 ~~ }
 
 -- | Access the 'realPart' of a 'Complex' number.
 --
@@ -117,7 +117,7 @@ _magnitude f c = setMag <$> f r
 
 -- | Access the 'phase' of a 'Complex' number.
 --
--- >>> (mkPolar 10 (2-pi) & _phase +~ pi & view _phase) ≈ 2
+-- >>> (mkPolar 10 (2-pi) & _phase +~ pi & view _phase) ~~ 2
 -- True
 --
 -- This isn't /quite/ a legal 'Lens'. Notably the
@@ -139,7 +139,7 @@ _phase f c = setPhase <$> f theta
 -- >>> (2.0 :+ 3.0) & _conjugate . _imagPart -~ 1
 -- 2.0 :+ 4.0
 --
--- >>> (mkPolar 10.0 2.0 ^. _conjugate . _phase) ≈ (-2.0)
+-- >>> (mkPolar 10.0 2.0 ^. _conjugate . _phase) ~~ (-2.0)
 -- True
 _conjugate :: RealFloat a => Iso' (Complex a) (Complex a)
 _conjugate = involuted conjugate
