Index: haskell-lens-4.1.2/lens.cabal
===================================================================
--- haskell-lens-4.1.2.orig/lens.cabal
+++ haskell-lens-4.1.2/lens.cabal
@@ -209,7 +209,6 @@ library
     template-haskell          >= 2.4      && < 2.11,
     text                      >= 0.11     && < 1.2,
     transformers              >= 0.2      && < 0.4,
-    transformers-compat       >= 0.1      && < 1,
     unordered-containers      >= 0.2      && < 0.3,
     utf8-string               >= 0.3.7    && < 0.4,
     vector                    >= 0.9      && < 0.11,
