--- a/yaml.cabal
+++ b/yaml.cabal
@@ -11,9 +11,6 @@
 cabal-version:   >= 1.2
 build-type:      Simple
 homepage:        http://github.com/snoyberg/yaml/
-extra-source-files: c/helper.h,
-                    c/yaml_private.h,
-                    c/yaml.h

 flag nolib
   description: Skip building the library
@@ -33,16 +30,7 @@
                    , enumerator >= 0.4 && < 0.5
     exposed-modules: Text.Libyaml
     ghc-options:     -Wall
-    c-sources:       c/helper.c,
-                     c/api.c,
-                     c/dumper.c,
-                     c/emitter.c,
-                     c/loader.c,
-                     c/parser.c,
-                     c/reader.c,
-                     c/scanner.c,
-                     c/writer.c
-    include-dirs:    c
+    extra-libraries: yaml

 executable           runtests
     if flag(buildtests)
@@ -60,14 +48,5 @@
     else
         Buildable: False
     ghc-options:     -Wall
-    c-sources:       c/helper.c,
-                     c/api.c,
-                     c/dumper.c,
-                     c/emitter.c,
-                     c/loader.c,
-                     c/parser.c,
-                     c/reader.c,
-                     c/scanner.c,
-                     c/writer.c
-    include-dirs:    c
+    extra-libraries: yaml
     main-is:         runtests.hs
