Description: Only compile with -msse2 on amd64/i386
Author: Colin Watson <cjwatson@debian.org>
Forwarded: no
Last-Update: 2015-05-08

Index: b/hmatrix.cabal
===================================================================
--- a/hmatrix.cabal
+++ b/hmatrix.cabal
@@ -90,7 +90,9 @@
                         -fno-warn-missing-signatures
                         -fno-warn-orphans

-    cc-options:         -O4 -msse2 -Wall
+    cc-options:         -O4 -Wall
+    if arch(x86_64) || arch(i386)
+        cc-options:     -msse2

     cpp-options:        -DBINARY
