--- a/notmuch-web.cabal
+++ b/notmuch-web.cabal
@@ -107,13 +107,13 @@

                      , aeson
                      , attoparsec
-                     , attoparsec-conduit
                      , blaze-builder
                      , blaze-html
                      , blaze-markup
                      , bytestring
                      , case-insensitive
                      , conduit
+                     , conduit-extra
                      , containers
                      , directory
                      , data-default
@@ -154,13 +154,13 @@

                      , aeson
                      , attoparsec
-                     , attoparsec-conduit
                      , blaze-builder
                      , blaze-html
                      , blaze-markup
                      , bytestring
                      , case-insensitive
                      , conduit
+                     , conduit-extra
                      , containers
                      , data-default
                      , directory
