Index: ghc-7.8.4/compiler/iface/MkIface.lhs
===================================================================
--- ghc-7.8.4.orig/compiler/iface/MkIface.lhs	2015-05-17 23:15:16.294100501 +0200
+++ ghc-7.8.4/compiler/iface/MkIface.lhs	2015-05-17 23:15:16.294100501 +0200
@@ -607,7 +607,7 @@
    iface_hash <- computeFingerprint putNameLiterally
                       (mod_hash,
                        ann_fn (mkVarOcc "module"),  -- See mkIfaceAnnCache
-                       mi_usages iface0,
+                       usages,
                        sorted_deps,
                        mi_hpc iface0)
 
@@ -639,6 +639,9 @@
     (non_orph_fis,   orph_fis)   = mkOrphMap ifFamInstOrph (mi_fam_insts iface0)
     fix_fn = mi_fix_fn iface0
     ann_fn = mkIfaceAnnCache (mi_anns iface0)
+    -- Do not allow filenames to affect the interface
+    usages = [ case u of UsageFile _ fp -> UsageFile "" fp; _ -> u | u <- mi_usages iface0 ]
+
 
 getOrphanHashes :: HscEnv -> [Module] -> IO [Fingerprint]
 getOrphanHashes hsc_env mods = do
