--- a/yesod-core.cabal
+++ b/yesod-core.cabal
@@ -60,7 +60,6 @@
                    , conduit               >= 1.0.11
                    , resourcet             >= 0.4.9    && < 1.2
                    , lifted-base           >= 0.1.2
-                   , attoparsec-conduit
                    , blaze-html            >= 0.5
                    , blaze-markup          >= 0.5.1
                    , data-default
@@ -125,7 +124,6 @@
                   , containers
                   , lifted-base
                   , resourcet
-                  , network-conduit
                   , network
                   , async
                   , conduit-extra
