--- a/yesod-core.cabal
+++ b/yesod-core.cabal
@@ -27,7 +27,7 @@
                    , time                  >= 1.1.4
                    , yesod-routes          >= 1.2      && < 1.3
                    , wai                   >= 1.4
-                   , wai-extra             >= 1.3
+                   , wai-extra             >= 3.0
                    , bytestring            >= 0.9.1.4
                    , text                  >= 0.7
                    , template-haskell
@@ -106,7 +106,6 @@
     cpp-options:   -DTEST
     build-depends: base
                   ,hspec >= 1.3
-                  ,wai-test >= 1.3.0.5
                   ,wai
                   ,yesod-core
                   ,bytestring
@@ -129,7 +128,7 @@
                   , conduit-extra
                   , shakespeare
                   , streaming-commons
-                  , wai-extra
+                  , wai-extra >= 3.0
     ghc-options:     -Wall
     extensions: TemplateHaskell
 
