Author: Sean Whitton <spwhitton@spwhitton.name>
Description: add missing test suite dependencies
Forwarded: peter.robinson@monoid.at

---
--- a/secret-sharing.cabal
+++ b/secret-sharing.cabal
@@ -65,6 +65,11 @@ test-suite Main
                    QuickCheck >= 2.4,
                    secret-sharing,
                    test-framework >= 0.4.1,
-                   test-framework-quickcheck2
+                   test-framework-quickcheck2,
+                   bytestring,
+                   vector,
+                   binary,
+                   finite-field,
+                   dice-entropy-conduit
   hs-source-dirs:  tests
   main-is:         Tests.hs
