---
 gloss.cabal |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: b/gloss.cabal
===================================================================
--- a/gloss.cabal
+++ b/gloss.cabal
@@ -37,7 +37,7 @@ Flag ExplicitBackend
 
 Library
   Build-Depends: 
-        base       >= 4.8 && < 4.10,
+        base       >= 4.8 && < 4.11,
         ghc-prim   >= 0.4 && < 0.6,
         containers == 0.5.*,
         bytestring == 0.10.*,
