--- a/hledger.cabal
+++ b/hledger.cabal
@@ -69,7 +69,6 @@ library
     , haskeline >= 0.6 && <= 0.8
     , HUnit
     , mtl
-    , mtl-compat
     , old-time
     , parsec >= 3
     , process
@@ -143,7 +142,6 @@ executable hledger
     , haskeline >= 0.6 && <= 0.8
     , HUnit
     , mtl
-    , mtl-compat
     , old-time
     , parsec >= 3
     , process
@@ -196,7 +194,6 @@ test-suite test
     , haskeline >= 0.6 && <= 0.8
     , HUnit
     , mtl
-    , mtl-compat
     , old-time
     , parsec >= 3
     , process
