--- a/hledger.cabal
+++ b/hledger.cabal
@@ -135,7 +135,6 @@
     , math-functions >=0.2.0.0
     , megaparsec >=7.0.0 && <8
     , mtl
-    , mtl-compat
     , old-time
     , parsec >=3
     , pretty-show >=1.6.4
@@ -186,7 +185,6 @@
     , math-functions >=0.2.0.0
     , megaparsec >=7.0.0 && <8
     , mtl
-    , mtl-compat
     , old-time
     , parsec >=3
     , pretty-show >=1.6.4
@@ -240,7 +238,6 @@
     , math-functions >=0.2.0.0
     , megaparsec >=7.0.0 && <8
     , mtl
-    , mtl-compat
     , old-time
     , parsec >=3
     , pretty-show >=1.6.4
@@ -295,7 +292,6 @@
     , math-functions >=0.2.0.0
     , megaparsec >=7.0.0 && <8
     , mtl
-    , mtl-compat
     , old-time
     , parsec >=3
     , pretty-show >=1.6.4
