--- a/hledger.cabal
+++ b/hledger.cabal
@@ -102,7 +102,6 @@ library
     , haskeline >=0.6 && <=0.8
     , HUnit
     , mtl
-    , mtl-compat
     , old-time
     , megaparsec >=5 && < 5.1
     , regex-tdfa
@@ -185,7 +184,6 @@ executable hledger
     , haskeline >=0.6 && <=0.8
     , HUnit
     , mtl
-    , mtl-compat
     , old-time
     , parsec >=3
     , regex-tdfa
@@ -246,7 +244,6 @@ test-suite test
     , haskeline >=0.6 && <=0.8
     , HUnit
     , mtl
-    , mtl-compat
     , old-time
     , parsec >=3
     , regex-tdfa
