--- a/hledger.cabal
+++ b/hledger.cabal
@@ -103,7 +103,6 @@ library
     , haskeline >=0.6 && <=0.8
     , HUnit
     , mtl
-    , mtl-compat
     , old-time
     , megaparsec >=5.0 && < 5.3
     , regex-tdfa
@@ -188,7 +187,6 @@ executable hledger
     , haskeline >=0.6 && <=0.8
     , HUnit
     , mtl
-    , mtl-compat
     , old-time
     , parsec >=3
     , regex-tdfa
@@ -250,7 +248,6 @@ test-suite test
     , haskeline >=0.6 && <=0.8
     , HUnit
     , mtl
-    , mtl-compat
     , old-time
     , parsec >=3
     , regex-tdfa
