--- a/hledger.cabal
+++ b/hledger.cabal
@@ -99,7 +99,6 @@
     , hledger-lib >=1.5 && <1.6
     , megaparsec >=5.0
     , mtl
-    , mtl-compat
     , old-time
     , pretty-show >=1.6.4
     , process
@@ -179,7 +178,6 @@
     , hledger
     , hledger-lib >=1.5 && <1.6
     , mtl
-    , mtl-compat
     , old-time
     , parsec >=3
     , pretty-show >=1.6.4
@@ -231,7 +229,6 @@
     , hledger
     , hledger-lib >=1.5 && <1.6
     , mtl
-    , mtl-compat
     , old-time
     , parsec >=3
     , pretty-show >=1.6.4
