--- a/hledger.cabal
+++ b/hledger.cabal
@@ -159,7 +159,6 @@
     , math-functions >=0.2.0.0
     , megaparsec >=7.0.0 && <8
     , mtl
-    , mtl-compat
     , old-time
     , parsec >=3
     , pretty-show >=1.6.4
@@ -208,7 +207,6 @@
     , math-functions >=0.2.0.0
     , megaparsec >=7.0.0 && <8
     , mtl
-    , mtl-compat
     , old-time
     , parsec >=3
     , pretty-show >=1.6.4
@@ -260,7 +258,6 @@
     , math-functions >=0.2.0.0
     , megaparsec >=7.0.0 && <8
     , mtl
-    , mtl-compat
     , old-time
     , parsec >=3
     , pretty-show >=1.6.4
@@ -313,7 +310,6 @@
     , math-functions >=0.2.0.0
     , megaparsec >=7.0.0 && <8
     , mtl
-    , mtl-compat
     , old-time
     , parsec >=3
     , pretty-show >=1.6.4
