Index: b/hledger.cabal
===================================================================
--- a/hledger.cabal
+++ b/hledger.cabal
@@ -132,7 +132,6 @@ library
     , lucid
     , megaparsec >=6.4.1
     , mtl
-    , mtl-compat
     , old-time
     , parsec >=3
     , pretty-show >=1.6.4
@@ -183,7 +182,6 @@ executable hledger
     , hledger-lib >=1.10 && <1.11
     , megaparsec >=6.4.1
     , mtl
-    , mtl-compat
     , old-time
     , parsec >=3
     , pretty-show >=1.6.4
@@ -237,7 +235,6 @@ test-suite test
     , hledger-lib >=1.10 && <1.11
     , megaparsec >=6.4.1
     , mtl
-    , mtl-compat
     , old-time
     , parsec >=3
     , pretty-show >=1.6.4
@@ -292,7 +289,6 @@ benchmark bench
     , html
     , megaparsec >=6.4.1
     , mtl
-    , mtl-compat
     , old-time
     , parsec >=3
     , pretty-show >=1.6.4
