--- a/Hledger/Cli/Version.hs
+++ b/Hledger/Cli/Version.hs
@@ -1,4 +1,4 @@
-{-# LANGUAGE CPP, TemplateHaskell #-}
+{-# LANGUAGE CPP #-}
 {-
 Version number-related utilities. See also the Makefile.
 -}
@@ -10,7 +10,6 @@
   binaryfilename
 )
 where
-import Distribution.PackageDescription.TH (packageVariable, package, pkgName, pkgVersion)
 import System.Info (os, arch)
 import Text.Printf

@@ -19,8 +18,8 @@

 -- package name and version from the cabal file
 progname, version, prognameandversion :: String
-progname = $(packageVariable (pkgName . package))
-version  = $(packageVariable (pkgVersion . package))
+progname = "hledger"
+version  = "0.17"
 prognameandversion = progname ++ " " ++ version

 -- developer build version strings include PATCHLEVEL (number of
--- a/hledger.cabal
+++ b/hledger.cabal
@@ -63,7 +63,6 @@
   build-depends:
                   hledger-lib == 0.17
                  ,base >= 3 && < 5
-                 ,cabal-file-th
                  ,containers
                  ,cmdargs >= 0.9.1   && < 0.10
                  ,csv
@@ -116,7 +115,6 @@
   build-depends:
                   hledger-lib == 0.17
                  ,base >= 3 && < 5
-                 ,cabal-file-th
                  ,containers
                  ,cmdargs >= 0.9.1   && < 0.10
                  ,csv
