--- a/config-value.cabal
+++ b/config-value.cabal
@@ -32,11 +32,11 @@
                        Config.Pretty,
                        Config.Value

-  build-depends:       base       >= 4.7     && < 4.9,
-                       array      >= 0.5     && < 0.6,
-                       bytestring >= 0.10.4  && < 0.11,
-                       pretty     >= 1.1.1.1 && < 1.2,
-                       text       >= 1.2.0.4 && < 1.3,
+  build-depends:       base       >= 4.6     && < 4.9,
+                       array      >= 0.4     && < 0.6,
+                       bytestring,
+                       pretty,
+                       text       >= 0.10.3  && < 1.3,
                        transformers >= 0.3   && < 0.5

   hs-source-dirs:      src
