--- a/hakyll.cabal
+++ b/hakyll.cabal
@@ -145,7 +145,7 @@
     citeproc-hs  >= 0.3.2  && < 0.4,
     cmdargs      >= 0.10   && < 0.11,
     containers   >= 0.3    && < 0.6,
-    cryptohash   >= 0.7    && < 0.10,
+    cryptohash   >= 0.7    && < 0.11,
     data-default >= 0.4    && < 0.6,
     deepseq      >= 1.3    && < 1.4,
     directory    >= 1.0    && < 1.3,
@@ -222,7 +222,7 @@
     citeproc-hs  >= 0.3.2  && < 0.4,
     cmdargs      >= 0.10   && < 0.11,
     containers   >= 0.3    && < 0.6,
-    cryptohash   >= 0.7    && < 0.10,
+    cryptohash   >= 0.7    && < 0.11,
     data-default >= 0.4    && < 0.6,
     deepseq      >= 1.3    && < 1.4,
     directory    >= 1.0    && < 1.3,
