--- a/tests/doctests.hsc
+++ b/tests/doctests.hsc
@@ -54,12 +54,12 @@
 main :: IO ()
 main = withUnicode $ getSources >>= \sources -> doctest $
     "-isrc"
-  : "-idist/build/autogen"
+  : "-idist-ghc/build/autogen"
   : "-optP-include"
-  : "-optPdist/build/autogen/cabal_macros.h"
+  : "-optPdist-ghc/build/autogen/cabal_macros.h"
   : "-hide-all-packages"
   : "-Iincludes"
-  : "dist/build/cbits/i2d.o"
+  : "dist-ghc/build/cbits/i2d.o"
   : map ("-package="++) (filter (not . ("bytes-" `isPrefixOf`)) deps) ++ sources
 
 getSources :: IO [FilePath]
