Index: b/src/Data/GADT/Compare.hs
===================================================================
--- a/src/Data/GADT/Compare.hs
+++ b/src/Data/GADT/Compare.hs
@@ -58,7 +58,7 @@ instance GRead ((:=) a) where
     greadsPrec p s = readsPrec p s >>= f
         where
             f :: forall x. (x := x, String) -> [(forall b. (forall a. x := a -> b) -> b, String)]
-            f (Refl, rest) = return ((\x -> x Refl) :: forall b. (forall a. x := a -> b) -> b, rest)
+            f (Refl, rest) = [((\x -> x Refl) :: forall b. (forall a. x := a -> b) -> b, rest)]
 
 -- |A class for type-contexts which contain enough information
 -- to (at least in some cases) decide the equality of types 
