Index: xmonad-0.9.1/XMonad/Core.hs
===================================================================
--- xmonad-0.9.1.orig/XMonad/Core.hs	2010-02-11 20:53:31.000000000 +0100
+++ xmonad-0.9.1/XMonad/Core.hs	2010-02-11 20:56:01.000000000 +0100
@@ -411,12 +411,18 @@
         base = dir </> "xmonad"
         err  = base ++ ".errors"
         src  = base ++ ".hs"
+	hi   = base ++ ".hi"
+	o    = base ++ ".o"
         lib  = dir </> "lib"
     libTs <- mapM getModTime . Prelude.filter isSource =<< allFiles lib
     srcT <- getModTime src
     binT <- getModTime bin
     if force || any (binT <) (srcT : libTs)
       then do
+        -- remove old interface files, just in case ghc6 was upgraded and doesn't
+	-- like the new ones. Ignore any errors while doing so.
+	try $ removeFile hi
+	try $ removeFile o
         -- temporarily disable SIGCHLD ignoring:
         uninstallSignalHandlers
         status <- bracket (openFile err WriteMode) hClose $ \h -> do
