Index: xmonad-0.10.1~darcs20120707/XMonad/Config.hs
===================================================================
--- xmonad-0.10.1~darcs20120707.orig/XMonad/Config.hs	2012-07-07 15:45:06.287889056 -0400
+++ xmonad-0.10.1~darcs20120707/XMonad/Config.hs	2012-07-07 15:45:08.123898152 -0400
@@ -151,7 +151,7 @@
 -- | The preferred terminal program, which is used in a binding below and by
 -- certain contrib modules.
 terminal :: String
-terminal = "xterm"
+terminal = "x-terminal-emulator"

 -- | Whether focus follows the mouse pointer.
 focusFollowsMouse :: Bool
