Index: bluetile-0.4.2/README
===================================================================
--- bluetile-0.4.2.orig/README	2010-06-28 09:50:15.000000000 +0200
+++ bluetile-0.4.2/README	2010-06-28 09:50:49.000000000 +0200
@@ -1,8 +1,7 @@
-Quickstart: After installing Bluetile (use 'cabal install') just open a
-terminal under your current window manager/desktop environment
-(preferably GNOME) and enter 'bluetile' or '/path/to/bluetile' if it's not
-in your PATH. Bluetile will replace the currently running window manager
-(if the window manager supports this) and start up.
+Quickstart: After installing Bluetile just open a terminal under your current
+window manager/desktop environment (preferably GNOME) and enter 'bluetile'.
+Bluetile will replace the currently running window manager (if the window
+manager supports this) and start up.

-See the man page of Bluetile for further information - including how to
-set up Bluetile as your permanent window manager.
+See the man page of Bluetile ("man bluetile") for further information -
+including how to set up Bluetile as your permanent window manager.
