Index: bluetile-0.5.3/etc/bluetilerc
===================================================================
--- bluetile-0.5.3.orig/etc/bluetilerc	2010-08-11 03:28:22.000000000 +0530
+++ bluetile-0.5.3/etc/bluetilerc	2011-04-20 22:13:55.000000000 +0530
@@ -8,7 +8,7 @@

 # global options (Mod4 is usually the logo key)
 default_modifier: Mod4
-terminal: gnome-terminal
+terminal: x-terminal-emulator
 start_dock: true

 # theme
