Index: bluetile-0.5/etc/bluetilerc
===================================================================
--- bluetile-0.5.orig/etc/bluetilerc	2010-07-26 19:49:35.000000000 +0200
+++ bluetile-0.5/etc/bluetilerc	2010-07-26 19:52:17.000000000 +0200
@@ -8,7 +8,7 @@

 # global options (Mod4 is usually the logo key)
 default_modifier: Mod4
-terminal: gnome-terminal
+terminal: x-terminal-emulator
 start_dock: true

 # launching and killing programs
