Index: haskell-hlist-0.2/HList.cabal
===================================================================
--- haskell-hlist-0.2.orig/HList.cabal	2010-04-21 20:11:23.000000000 +0200
+++ haskell-hlist-0.2/HList.cabal	2010-04-21 20:11:29.000000000 +0200
@@ -25,5 +25,5 @@

 extensions:          EmptyDataDecls, FlexibleContexts, FlexibleInstances, FunctionalDependencies,
                      MultiParamTypeClasses, OverlappingInstances, ScopedTypeVariables, RankNTypes,
-                     ScopedTypeVariables, TypeSynonymInstances, UndecidableInstances
+                     ScopedTypeVariables, TypeSynonymInstances, UndecidableInstances, TypeOperators
 ghc-options:         -Wall
