--- a/rss2irc.cabal
+++ b/rss2irc.cabal
@@ -103,7 +103,7 @@
                      irc                   >= 0.4 && < 0.5,
                      mtl                   >= 1.1,
                      network               >= 2.2,
-                     parallel              >= 1.1 && < 1.2,
+                     deepseq               >= 1.1,
                      regexpr               >= 0.5 && < 0.6,
                      split                 >= 0.1 && < 0.2,
                      strict-concurrency    >= 0.2 && < 0.3,
--- a/rss2irc.hs
+++ b/rss2irc.hs
@@ -27,7 +27,7 @@
 import Network.URI (URI(URI),uriScheme,parseURI)
 import qualified Control.Exception.Extensible as E (bracket,catch)
 import Control.Exception.Extensible (fromException)
-import Control.Parallel.Strategies (NFData) --(rnf),($|))
+import Control.DeepSeq (NFData)
 import System.Console.GetOpt (OptDescr(Option), ArgDescr(ReqArg,NoArg), ArgOrder(Permute), getOpt, usageInfo)
 import System.Environment (getArgs)
 import System.Exit (exitWith, ExitCode(ExitSuccess), exitFailure)
