--- a/rss2irc.cabal
+++ b/rss2irc.cabal
@@ -100,7 +100,7 @@
                      feed                  >= 0.3 && < 0.4,
                      haskell98,
                      HTTP                  >= 4000.0.3,
-                     irc                   >= 0.4 && < 0.5,
+                     irc                   >= 0.4 && < 0.6,
                      mtl                   >= 1.1,
                      network               >= 2.2,
                      deepseq               >= 1.1,
