--- a/rss2irc.cabal
+++ b/rss2irc.cabal
@@ -99,7 +99,7 @@
                      extensible-exceptions >= 0.1 && < 0.2,
                      feed                  >= 0.3 && < 0.4,
                      haskell98,
-                     HTTP                  >= 4000.0.3 && < 4000.1,
+                     HTTP                  >= 4000.0.3,
                      irc                   >= 0.4 && < 0.5,
                      mtl                   >= 1.1 && < 1.2,
                      network               >= 2.2 && < 2.3,
