--- a/rss2irc.cabal
+++ b/rss2irc.cabal
@@ -98,11 +98,11 @@
                      base                  >= 3 && < 5,
                      extensible-exceptions >= 0.1 && < 0.2,
                      feed                  >= 0.3 && < 0.4,
-                     haskell98,
                      HTTP                  >= 4000.0.3,
                      irc                   >= 0.4 && < 0.6,
                      mtl                   >= 1.1,
                      network               >= 2.2,
+                     old-locale            >= 1.0.0.4,
                      deepseq               >= 1.1,
                      regexpr               >= 0.5 && < 0.6,
                      split                 >= 0.1 && < 0.2,
--- a/rss2irc.hs
+++ b/rss2irc.hs
@@ -19,7 +19,7 @@
 import Data.Time.Clock (UTCTime,getCurrentTime,diffUTCTime)
 import Data.Time.Format (parseTime)
 import Data.Time.LocalTime (LocalTime,getCurrentTimeZone,utcToLocalTime)
-import Locale (defaultTimeLocale)
+import System.Locale (defaultTimeLocale)
 import Network (PortID(PortNumber), connectTo)
 import Network.Browser (browse,request,setAllowRedirects,setOutHandler)
 import Network.HTTP (Response,getRequest,rspBody,rspCode)
