From a627b4413773b33fe5db31717c85c67bfee95cac Mon Sep 17 00:00:00 2001
From: Brent Yorgey <byorgey@gmail.com>
Date: Wed, 25 Oct 2017 11:55:27 -0500
Subject: [PATCH] allow vector-space-0.12

---
 vector-space-points.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vector-space-points.cabal b/vector-space-points.cabal
index 9ce2504..b07e5fe 100644
--- a/vector-space-points.cabal
+++ b/vector-space-points.cabal
@@ -19,6 +19,6 @@ source-repository head
 library
   exposed-modules:     Data.AffineSpace.Point
   build-depends:       base >=4.0 && < 4.11,
-                       vector-space >=0.7 && < 0.12
+                       vector-space >=0.7 && < 0.13
   hs-source-dirs:      src
   default-language:    Haskell2010
