From d862832824aa4f69615c24c27cb0c16246cc5503 Mon Sep 17 00:00:00 2001
From: Brent Yorgey <byorgey@gmail.com>
Date: Tue, 8 Aug 2017 15:28:23 -0500
Subject: [PATCH] allow base-4.10 and vector-space-0.11

---
 vector-space-points.cabal | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/vector-space-points.cabal b/vector-space-points.cabal
index 10cd74d..9ce2504 100644
--- a/vector-space-points.cabal
+++ b/vector-space-points.cabal
@@ -18,7 +18,7 @@ source-repository head
 
 library
   exposed-modules:     Data.AffineSpace.Point
-  build-depends:       base >=4.0 && < 4.10,
-                       vector-space >=0.7 && < 0.11
+  build-depends:       base >=4.0 && < 4.11,
+                       vector-space >=0.7 && < 0.12
   hs-source-dirs:      src
   default-language:    Haskell2010
