From d87a4166c778fa4b3410d8f9838b34668bc1684c Mon Sep 17 00:00:00 2001
From: Brent Yorgey <byorgey@gmail.com>
Date: Mon, 19 Feb 2018 07:11:31 -0600
Subject: [PATCH] allow vector-space-0.13

---
 vector-space-points.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vector-space-points.cabal b/vector-space-points.cabal
index b07e5fe..f2f399f 100644
--- a/vector-space-points.cabal
+++ b/vector-space-points.cabal
@@ -19,6 +19,6 @@ source-repository head
 library
   exposed-modules:     Data.AffineSpace.Point
   build-depends:       base >=4.0 && < 4.11,
-                       vector-space >=0.7 && < 0.13
+                       vector-space >=0.7 && < 0.14
   hs-source-dirs:      src
   default-language:    Haskell2010
