--- a/hledger-vty.cabal
+++ b/hledger-vty.cabal
@@ -37,8 +37,8 @@
                   Hledger.Vty.Main
                   Hledger.Vty.Options
   build-depends:
-                  hledger == 0.16.1
-                 ,hledger-lib == 0.16.1
+                  hledger == 0.17
+                 ,hledger-lib == 0.17
                  ,HUnit
                  ,base >= 3 && < 5
                  ,cmdargs >= 0.8   && < 0.10
--- a/Hledger/Vty/Options.hs
+++ b/Hledger/Vty/Options.hs
@@ -7,11 +7,11 @@
 import System.Console.CmdArgs
 import System.Console.CmdArgs.Explicit

-import Hledger.Cli hiding (progname,progversion)
+import Hledger.Cli hiding (progname)
 import qualified Hledger.Cli (progname)

 progname = Hledger.Cli.progname ++ "-vty"
-progversion = progversionstr progname
+progversion = "0.16.1debian"

 vtyflags = [
   flagNone ["debug-vty"]  (\opts -> setboolopt "rules-file" opts) "run with no terminal output, showing console"
--- a/Hledger/Vty/Main.hs
+++ b/Hledger/Vty/Main.hs
@@ -16,7 +16,7 @@
 import Text.Printf

 import Hledger
-import Hledger.Cli hiding (progname,progversion)
+import Hledger.Cli hiding (progname)
 import Hledger.Vty.Options
 import Prelude hiding (putStrLn)
 import Hledger.Utils.UTF8 (putStrLn)
