--- a/mighttpd2.cabal
+++ b/mighttpd2.cabal
@@ -21,7 +21,7 @@
                       , deepseq
                       , directory
                       , filepath
-                      , hashmap >= 1.2 && < 1.3
+                      , hashmap >= 1.2 && < 1.4
                       , http-conduit
                       , http-date
                       , http-types
@@ -65,7 +65,7 @@
                       , deepseq
                       , directory
                       , filepath
-                      , hashmap >= 1.2 && < 1.3
+                      , hashmap >= 1.2 && < 1.4
                       , http-conduit
                       , http-date
                       , http-types
