--- haskell-filestore-0.6.2.orig/filestore.cabal
+++ haskell-filestore-0.6.2/filestore.cabal
@@ -33,7 +33,7 @@ Library
                          containers >= 0.3 && < 0.6,
                          utf8-string >= 0.3 && < 1.1,
                          filepath >= 1.1 && < 1.5,
-                         directory >= 1.0 && < 1.3,
+                         directory >= 1.0 && < 1.4,
                          parsec >= 2 && < 3.2,
                          process >= 1.0 && < 1.5,
                          time >= 1.1 && < 1.7,
@@ -70,6 +70,6 @@ Test-suite test-filestore
                     time,
                     Diff >= 0.2 && < 0.4,
                     filepath >= 1.1 && < 1.5,
-                    directory >= 1.1 && < 1.3,
+                    directory >= 1.1 && < 1.4,
                     filestore
 
