Index: haskell-cabal-1.20.0.1/Cabal.cabal
===================================================================
--- haskell-cabal-1.20.0.1.orig/Cabal.cabal	2014-07-30 22:51:39.555791255 +0200
+++ haskell-cabal-1.20.0.1/Cabal.cabal	2014-07-30 22:51:39.555791255 +0200
@@ -238,7 +238,7 @@
     test-framework-hunit,
     test-framework-quickcheck2,
     HUnit,
-    QuickCheck < 2.7,
+    QuickCheck < 2.8,
     Cabal
   ghc-options: -Wall
   default-language: Haskell98
@@ -283,7 +283,7 @@
     test-framework-quickcheck2 >= 0.2.12,
     test-framework-hunit,
     HUnit,
-    QuickCheck >= 2.1.0.1 && < 2.7,
+    QuickCheck >= 2.1.0.1 && < 2.8,
     Cabal,
     process,
     directory,
