--- haskell-yesod-auth-oauth2-0.2.2.orig/yesod-auth-oauth2.cabal
+++ haskell-yesod-auth-oauth2-0.2.2/yesod-auth-oauth2.cabal
@@ -31,7 +31,7 @@ library
                    , http-client             >= 0.4.0     && < 0.5.8
                    , http-conduit            >= 2.0       && < 3.0
                    , http-types              >= 0.8       && < 0.10
-                   , aeson                   >= 0.6       && < 0.12
+                   , aeson                   >= 0.6       && < 1.0.2.2
                    , yesod-core              >= 1.2       && < 1.5
                    , authenticate            >= 1.3.2.7   && < 1.4
                    , random
