Last-Update: 2017-07-22

--- haskell-yesod-auth-oauth2-0.2.2.orig/yesod-auth-oauth2.cabal
+++ haskell-yesod-auth-oauth2-0.2.2/yesod-auth-oauth2.cabal
@@ -28,7 +28,7 @@ library
 
     build-depends:   base                    >= 4.5       && < 5
                    , bytestring              >= 0.9.1.4
-                   , http-client             >= 0.4.0     && < 0.5
+                   , http-client             >= 0.4.0     && < 0.5.8
                    , http-conduit            >= 2.0       && < 3.0
                    , http-types              >= 0.8       && < 0.10
                    , aeson                   >= 0.6       && < 0.12
