From a35b5570d6d729a38d68e309c3bd92f3138e3833 Mon Sep 17 00:00:00 2001
From: Bryan O'Sullivan <bos@serpentine.com>
Date: Thu, 21 Jan 2016 22:24:48 -0800
Subject: [PATCH] Fix goodProducer on 32-bit systems

---
 tests/UnitTests.hs | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

Index: haskell-aeson/tests/UnitTests.hs
===================================================================
--- haskell-aeson.orig/tests/UnitTests.hs	2016-01-22 22:07:57.924492201 +0100
+++ haskell-aeson/tests/UnitTests.hs	2016-01-22 22:07:57.920492118 +0100
@@ -77,9 +77,12 @@
                            '{' (L.head (encode wibble))
   where
     wibble = Wibble {
-                 wibbleString = replicate 4030 'a'
-               , wibbleInt = undefined
+                 wibbleString = replicate k 'a'
+               , wibbleInt = 1
                }
+    k | arch32bit = 4047
+      | otherwise = 4030
+    arch32bit     = (maxBound :: Int) == 2147483647
 
 -- Test decoding various UTC time formats
 --
