--- a/aeson.cabal
+++ b/aeson.cabal
@@ -81,6 +81,10 @@
   description: Use the new bytestring builder available in bytestring >= 0.10.4.0
   default: False

+flag th
+  description: build TH bits
+  default: True
+
 library
   exposed-modules:
     Data.Aeson
@@ -88,7 +92,8 @@
     Data.Aeson.Generic
     Data.Aeson.Parser
     Data.Aeson.Types
-    Data.Aeson.TH
+  if flag(th)
+      exposed-modules: Data.Aeson.TH

   other-modules:
     Data.Aeson.Functions
