--- a/aeson.cabal
+++ b/aeson.cabal
@@ -62,6 +62,10 @@
                If true then depend on time < 1.5 together with old-locale.
   default: False
 
+flag th
+  description: build TH bits
+  default: True
+
 library
   exposed-modules:
     Data.Aeson
@@ -70,7 +74,8 @@
     Data.Aeson.Internal.Time
     Data.Aeson.Parser
     Data.Aeson.Types
-    Data.Aeson.TH
+  if flag(th)
+      exposed-modules: Data.Aeson.TH
 
   other-modules:
     Data.Aeson.Encode.Builder
