--- a/aeson.cabal
+++ b/aeson.cabal
@@ -95,6 +95,10 @@
   description: build TH bits
   default: True

+flag threaded
+  description: build threaded
+  default: True
+
 library
   exposed-modules:
     Data.Aeson
@@ -152,11 +156,13 @@
                   Properties.Deprecated

   ghc-options:
-    -Wall -threaded -rtsopts
+    -Wall -rtsopts
   if impl(ghc < 7.4)
     ghc-options: -fcontext-stack=40
   if impl(ghc >= 7.2)
     cpp-options: -DGHC_GENERICS
+  if flag(threaded)
+    ghc-options: -threaded

   build-depends:
     QuickCheck,
