--- a/aeson.cabal
+++ b/aeson.cabal
@@ -68,6 +68,10 @@ flag th
   description: build TH bits
   default: True
 
+flag threaded
+  description: build threaded
+  default: True
+
 library
   default-language: Haskell2010
 
@@ -132,6 +136,10 @@ library
 
   include-dirs: include
 
+  if flag(threaded)
+    ghc-options: -threaded
+
+
 test-suite tests
   default-language: Haskell2010
   type:           exitcode-stdio-1.0
