Author: Gianfranco Costamagna <locutusofborg@debian.org>
Last-Update: 2017-07-22

--- a/xmonad-wallpaper.cabal
+++ b/xmonad-wallpaper.cabal
@@ -59,7 +59,7 @@ library
   -- other-extensions:
 
   -- Other library packages from which modules are imported.
-  build-depends:       base >=3.0 && <5.0, unix, xmonad >=0.13 && <0.16, mtl, random, magic >=1.1 && <1.2
+  build-depends:       base >=3.0 && <5.0, unix, xmonad >=0.13 && <0.18, mtl, random, magic >=1.1 && <1.2
 
   -- Directories containing source files.
   hs-source-dirs:      src
