Index: haskell-hslua/hslua.cabal
===================================================================
--- haskell-hslua.orig/hslua.cabal	2015-08-15 14:37:56.512810553 +0200
+++ haskell-hslua/hslua.cabal	2015-08-15 14:39:58.063289762 +0200
@@ -30,7 +30,7 @@
 
 flag system-lua
   description:          Use the system-wide Lua instead of the bundled copy
-  default:              False
+  default:              True
 
 flag apicheck
   description:          Compile Lua with -DLUA_USE_APICHECK.
@@ -49,7 +49,7 @@
     if flag(luajit)
       Extra-libraries:  luajit-5.1
     else
-      Extra-libraries:  lua
+      Extra-libraries:  lua5
     includes:           lua.h
   else
     c-sources:          lua-5.1.5/lobject.c, lua-5.1.5/ltm.c, lua-5.1.5/ldblib.c,
