Description: Build with ld --as-needed
Author: Matthias Klose <doko@ubuntu.com>
Forwarded: yes, https://github.com/bos/llvm/pull/1

Index: haskell-llvm/configure
===================================================================
--- haskell-llvm.orig/configure	2011-05-28 09:39:00.248596000 +0100
+++ haskell-llvm/configure	2011-05-28 09:39:45.202435000 +0100
@@ -3996,7 +3996,7 @@
 done


-LIBS="-lLLVMSupport -lLLVMSystem $LIBS"
+LIBS="-lLLVMSupport -lLLVMSystem -lpthread -ldl $LIBS"

 # We have to link using the C++ compiler.
 CC=$CXX
Index: haskell-llvm/configure.ac
===================================================================
--- haskell-llvm.orig/configure.ac	2011-05-28 09:39:00.248596000 +0100
+++ haskell-llvm/configure.ac	2011-05-28 09:39:00.248596000 +0100
@@ -81,7 +81,7 @@
 AC_CHECK_HEADERS([llvm-c/Core.h], [],
   [AC_MSG_ERROR(could not find LLVM C bindings)])

-LIBS="-lLLVMSupport -lLLVMSystem $LIBS"
+LIBS="-lLLVMSupport -lLLVMSystem -lpthread -ldl $LIBS"

 # We have to link using the C++ compiler.
 CC=$CXX
