Index: gtk2hs-buildtools-0.12.0/c2hs/c/CLexer.x
===================================================================
--- gtk2hs-buildtools-0.12.0.orig/c2hs/c/CLexer.x	2011-09-06 08:42:27.000000000 +0200
+++ gtk2hs-buildtools-0.12.0/c2hs/c/CLexer.x	2011-09-06 08:42:32.000000000 +0200
@@ -78,6 +78,8 @@
 import CTokens
 import CParserMonad

+import Data.Word (Word8)
+
 }

 $space = [ \ \t ] -- horizontal white space
@@ -390,6 +392,12 @@
 alexInputPrevChar :: AlexInput -> Char
 alexInputPrevChar _ = error "alexInputPrevChar not used"

+-- for alex 3
+alexGetByte :: AlexInput -> Maybe (Word8, AlexInput)
+alexGetByte (p,[]) = Nothing
+alexGetByte (p, (c:s)) = let p' = alexMove p c in p' `seq`
+                           Just ((fromIntegral $ ord c), (p', s))
+
 alexGetChar :: AlexInput -> Maybe (Char,AlexInput)
 alexGetChar (p,[]) = Nothing
 alexGetChar (p,(c:s))  = let p' = alexMove p c in p' `seq`
