Forwarded: http://code.google.com/p/ndmitchell/issues/detail?id=590

Index: haskell-hoogle-4.2.10/src/CmdLine/Type.hs
===================================================================
--- haskell-hoogle-4.2.10.orig/src/CmdLine/Type.hs	2012-09-13 09:59:22.839204283 +0900
+++ haskell-hoogle-4.2.10/src/CmdLine/Type.hs	2012-10-12 00:42:15.267088185 +0900
@@ -34,7 +34,7 @@
     | Data {redownload :: Bool, local :: [String], datadir :: FilePath, threads :: Int, actions :: [String]}
     | Server {port :: Int, local_ :: Bool, databases :: [FilePath], resources :: FilePath, dynamic :: Bool, template :: [FilePath]}
     | Combine {srcfiles :: [FilePath], outfile :: String}
-    | Convert {srcfile :: String, outfile :: String}
+    | Convert {srcfile :: String, outfile :: String, addlocation :: Bool}
     | Log {logfiles :: [FilePath]}
     | Test {testFiles :: [String], example :: Bool}
     | Dump {database :: String, section :: [String]}
@@ -92,6 +92,7 @@
 convert = Convert
     {srcfile = def &= argPos 0 &= typ "INPUT"
     ,outfile = def &= argPos 1 &= typ "DATABASE" &= opt ""
+    ,addlocation = def &= help "Add location infomation to database"
     } &= help "Convert an input file to a database"

 data_ = Data
Index: haskell-hoogle-4.2.10/src/Console/All.hs
===================================================================
--- haskell-hoogle-4.2.10.orig/src/Console/All.hs	2012-09-13 09:59:22.839204283 +0900
+++ haskell-hoogle-4.2.10/src/Console/All.hs	2012-10-12 00:43:26.363440741 +0900
@@ -3,6 +3,7 @@

 import CmdLine.All
 import Recipe.All
+import Recipe.Hackage
 import Console.Log
 import Console.Search
 import Console.Test
@@ -36,12 +37,15 @@

 action (Log files) = logFiles files

-action (Convert from to) = do
+action (Convert from to addloc) = do
     to <- return $ if null to then replaceExtension from "hoo" else to
     when (any isUpper $ takeBaseName to) $ putStrLn $ "Warning: Hoogle databases should be all lower case, " ++ takeBaseName to
     putStrLn $ "Converting " ++ from
     src <- readFileUtf8 from
-    let (err,db) = createDatabase Haskell [] src
+    let from' = "file:/" ++ from
+        src' = if addloc then unlines . haddockPackageUrl from' . lines $ src
+               else src
+        (err,db) = createDatabase Haskell [] src'
     unless (null err) $ putStr $ unlines $ "Warning: parse errors" : map show err
     saveDatabase to db
     putStrLn $ "Written " ++ to
Index: haskell-hoogle-4.2.10/src/Recipe/Hackage.hs
===================================================================
--- haskell-hoogle-4.2.10.orig/src/Recipe/Hackage.hs	2012-09-13 09:59:22.839204283 +0900
+++ haskell-hoogle-4.2.10/src/Recipe/Hackage.hs	2012-10-12 00:42:15.271088200 +0900
@@ -1,5 +1,5 @@

-module Recipe.Hackage(makePlatform, makeDefault, makePackage, makeAll) where
+module Recipe.Hackage(makePlatform, makeDefault, makePackage, makeAll, haddockPackageUrl) where

 import Recipe.Type
 import Recipe.Cabal
