--- a/hbro.cabal
+++ b/hbro.cabal
@@ -61,13 +61,33 @@

 Executable hbro
     Build-depends:
-        hbro,
         base == 4.*,
         directory,
         mtl
+      , bytestring,
+        cmdargs,
+        containers,
+        dyre,
+        filepath,
+        glib,
+        gtk,
+        network,
+        pango,
+        process,
+        transformers,
+        unix,
+        webkit,
+        xdg-basedir,
+        zeromq-haskell
     Main-is: Main.hs
-    Hs-Source-Dirs: Hbro
+    Hs-Source-Dirs: Hbro, .
     Ghc-options: -Wall
     if flag(threaded)
         Ghc-options: -threaded
-
+    other-modules:
+        Hbro.Core,
+        Hbro.Hbro,
+        Hbro.Gui,
+        Hbro.Socket,
+        Hbro.Types,
+        Hbro.Util
