Description: Fix type error of unknown source
Author: Dmitry Bogatov <KAction@gnu.org>
Forwarded: yes
Last-Update: 2014-12-04
--- a/src/Text/BlogLiterately/Transform.hs
+++ b/src/Text/BlogLiterately/Transform.hs
@@ -367,7 +367,7 @@ xformDoc bl xforms =
                    }
     mathOption opt
       | opt `isPrefixOf` "latexmathml" ||
-        opt `isPrefixOf` "asciimathml" = LaTeXMathML (mathUrlMaybe opt)
+        opt `isPrefixOf` "asciimathml" = LaTeXMathML (urlPart opt)
       | opt `isPrefixOf` "mathml"      = MathML (mathUrlMaybe opt)
       | opt `isPrefixOf` "mimetex"     =
           WebTeX (mathUrl "/cgi-bin/mimetex.cgi?" opt)
