Description: Remove windows-used mintty dependency
Author: Gianfranco Costamagna <locutusofborg@debian.org>
Last-Update: 2018-01-03

--- haskell-stack-1.5.1.orig/stack.cabal
+++ haskell-stack-1.5.1/stack.cabal
@@ -231,7 +231,6 @@ library
                    , memory >= 0.13 && < 0.15
                    , microlens >= 0.3.0.0
                    , microlens-mtl >= 0.1.10.0
-                   , mintty >= 0.1.1
                    , monad-control >= 1.0.1.0
                    , monad-logger >= 0.3.13.1
                    , monad-unlift < 0.3
