Description: Remove windows-used mintty dependency
Author: Gianfranco Costamagna <locutusofborg@debian.org>
Last-Update: 2018-01-03

Index: b/stack.cabal
===================================================================
--- a/stack.cabal
+++ b/stack.cabal
@@ -255,7 +255,6 @@ library
     , http-types
     , memory
     , microlens
-    , mintty
     , monad-logger
     , mono-traversable
     , mtl
@@ -363,7 +362,6 @@ executable stack
     , http-types
     , memory
     , microlens
-    , mintty
     , monad-logger
     , mono-traversable
     , mtl
@@ -482,7 +480,6 @@ test-suite stack-integration-test
     , http-types
     , memory
     , microlens
-    , mintty
     , monad-logger
     , mono-traversable
     , mtl
@@ -605,7 +602,6 @@ test-suite stack-test
     , http-types
     , memory
     , microlens
-    , mintty
     , monad-logger
     , mono-traversable
     , mtl
