Description: Remove windows-used mintty dependency
Author: Gianfranco Costamagna <locutusofborg@debian.org>
Last-Update: 2018-01-03

Index: b/stack.cabal
===================================================================
--- a/stack.cabal
+++ b/stack.cabal
@@ -307,7 +307,6 @@ library
     , http-types
     , memory
     , microlens
-    , mintty
     , mono-traversable
     , mtl
     , mustache
@@ -432,7 +431,6 @@ executable stack
     , http-types
     , memory
     , microlens
-    , mintty
     , mono-traversable
     , mtl
     , mustache
@@ -556,7 +554,6 @@ executable stack-integration-test
     , http-types
     , memory
     , microlens
-    , mintty
     , mono-traversable
     , mtl
     , mustache
@@ -687,7 +684,6 @@ test-suite stack-test
     , http-types
     , memory
     , microlens
-    , mintty
     , mono-traversable
     , mtl
     , mustache
