Description: Remove windows-used mintty dependency
Author: Gianfranco Costamagna <locutusofborg@debian.org>
Last-Update: 2018-01-03

--- a/stack.cabal
+++ b/stack.cabal
@@ -130,7 +130,6 @@
     , memory
     , microlens
     , microlens-mtl
-    , mintty
     , monad-logger
     , mono-traversable
     , mtl
@@ -365,7 +364,6 @@
     , memory
     , microlens
     , microlens-mtl
-    , mintty
     , monad-logger
     , mono-traversable
     , mtl
@@ -483,7 +481,6 @@
     , memory
     , microlens
     , microlens-mtl
-    , mintty
     , monad-logger
     , mono-traversable
     , mtl
@@ -592,7 +589,6 @@
     , memory
     , microlens
     , microlens-mtl
-    , mintty
     , monad-logger
     , mono-traversable
     , mtl
