Description: Maps dependencies to lowercase.
Author: Max Kirillov <max@max630.net>
Reviewed-by: Sven Bartscher <sven.bartscher@weltraumschlangen.de>
Last-Update: 2014-07-30
Forwarded: Yes
--- a/src/Debian/Debianize/BuildDependencies.hs
+++ b/src/Debian/Debianize/BuildDependencies.hs
@@ -7,7 +7,7 @@

 import Control.Monad.State (MonadState(get))
 import Control.Monad.Trans (MonadIO)
-import Data.Char (isSpace)
+import Data.Char (isSpace, toLower)
 import Data.Function (on)
 import Data.Lens.Lazy (access, getL)
 import Data.List as List (filter, map, minimumBy, nub)
@@ -75,7 +75,7 @@
     where
       fixDeps :: Atoms -> [String] -> Relations
       fixDeps atoms xs =
-          concatMap (\ cab -> fromMaybe [[D.Rel (D.BinPkgName ("lib" ++ cab ++ "-dev")) Nothing Nothing]]
+          concatMap (\ cab -> fromMaybe [[D.Rel (D.BinPkgName ("lib" ++ map toLower cab ++ "-dev")) Nothing Nothing]]
                                         (Map.lookup cab (getL T.extraLibMap atoms))) xs

 -- The haskell-cdbs package contains the hlibrary.mk file with
