Index: haskell-hashed-storage-0.3.9/Storage/Hashed/Index.hs
===================================================================
--- haskell-hashed-storage-0.3.9.orig/Storage/Hashed/Index.hs	2010-02-25 09:57:07.000000000 +0100
+++ haskell-hashed-storage-0.3.9/Storage/Hashed/Index.hs	2010-02-25 09:58:09.000000000 +0100
@@ -187,8 +187,8 @@
                  if req_size > 0 then fromIntegral req_size else act_size
   case size of
     0 -> return (castForeignPtr nullForeignPtr, size)
-    _ -> do (x, _) <- mmapFileForeignPtr indexpath
-                                         ReadWrite (Just (0, size + 4))
+    _ -> do (x, _, _) <- mmapFileForeignPtr indexpath
+                                            ReadWrite (Just (0, size + 4))
             return (x, size)

 -- | See 'readIndex'. This version also gives a map from paths to items, so the
Index: haskell-hashed-storage-0.3.9/hashed-storage.cabal
===================================================================
--- haskell-hashed-storage-0.3.9.orig/hashed-storage.cabal	2010-02-25 09:57:06.000000000 +0100
+++ haskell-hashed-storage-0.3.9/hashed-storage.cabal	2010-02-25 09:57:16.000000000 +0100
@@ -60,7 +60,7 @@
         Storage.Hashed.Utils

     build-depends: base >= 3 && < 5, directory, filepath, bytestring, zlib,
-                   containers, mtl, extensible-exceptions, mmap >= 0.4 && < 0.5
+                   containers, mtl, extensible-exceptions, mmap >= 0.5 && < 0.6

     c-sources: Bundled/sha2.c
