--- a/Setup.hs
+++ b/Setup.hs
@@ -16,6 +16,7 @@
 import Distribution.Simple.Utils
 import Distribution.Text
 import Distribution.Verbosity
+import qualified Data.Version as V
 import qualified Distribution.InstalledPackageInfo as I
 import qualified Distribution.ModuleName as ModuleName
 import Distribution.PackageDescription
@@ -237,6 +238,8 @@
       dist    = fromFlag $ regDistPref flags
       pkgDb   = withPackageDB lbi
       clbi    = extractCLBI lbi
+      PackageName name = packageName pkg
+      version = packageVersion pkg
   instPkgInfo <- generateRegistrationInfo
     verb pkg lib lbi clbi inplace dist
   let instPkgInfo' = instPkgInfo {
@@ -252,6 +255,7 @@
     Just regFile -> do
       writeUTF8File (fromMaybe (display (packageId pkg) <.> "conf") regFile) $
         I.showInstalledPackageInfo instPkgInfo'
+      putStrLn ("Creating package registration file: " ++ name ++ "-" ++ V.showVersion version ++ ".conf")
     _ | fromFlag (regGenScript flags) ->
       die "Registration scripts are not implemented."
       | otherwise ->
