--- a/Setup.hs
+++ b/Setup.hs
@@ -8,12 +8,4 @@ import System.Exit
 import System.FilePath ((</>))
 import System.Process

-main = copyFile "debian/changelog" "changelog" >>
-       defaultMainWithHooks simpleUserHooks {
-         postBuild = \ _ _ _ lbi -> runTestScript lbi
-       , runTests = \ _ _ _ lbi -> runTestScript lbi
-       }
-
-runTestScript lbi =
-    system (buildDir lbi </> "debian-tests/debian-tests") >>= \ code ->
-    if code == ExitSuccess then return () else error "Test Failure"
+main = defaultMain
--- a/debian.cabal
+++ b/debian.cabal
@@ -93,11 +93,7 @@ Library
         Debian.UTF8,
         Debian.Util.FakeChanges
  other-modules:
-        Debian.Version.Internal,
-        Test.Changes,
-        Test.Dependencies,
-        Test.SourcesList,
-        Test.Versions
+        Debian.Version.Internal

 Executable fakechanges
  Hs-Source-Dirs: utils
@@ -124,15 +120,6 @@ Executable apt-get-build-depends
  ghc-options: -threaded -W -O2
  Extensions:           ExistentialQuantification CPP

-Executable debian-tests
- Main-is: Test/Main.hs
- ghc-options: -threaded -W -O2
- Build-Depends: ansi-wl-pprint, base, bytestring, containers, debian, directory, exceptions, HUnit, ListLike >= 4.1.1, mtl, parsec, pretty, prettyclass, process, process-extras >= 0.2.0, regex-compat, regex-tdfa, template-haskell, text, utf8-string
- if flag(network-uri)
-   Build-Depends: network >= 2.6, network-uri >= 2.6
- else
-   Build-Depends: network >= 2.4 && < 2.6
-
 source-repository head
   type:     git
   location: https://github.com/ddssff/debian-haskell
