--- a/debian.cabal
+++ b/debian.cabal
@@ -33,7 +33,6 @@
    directory,
    filepath,
    HaXml >= 1.20,
-   HUnit,
    mtl,
    network >= 2.4,
    old-locale,
@@ -95,11 +94,7 @@
         Debian.UTF8,
         Debian.Util.FakeChanges
  other-modules:
-        Debian.Version.Internal,
-        Test.Changes,
-        Test.Dependencies,
-        Test.SourcesList,
-        Test.Versions
+        Debian.Version.Internal

 Executable fakechanges
  Main-is: utils/FakeChanges.hs
--- a/Setup.hs
+++ b/Setup.hs
@@ -6,8 +6,7 @@
 import System.Exit

 main = defaultMainWithHooks simpleUserHooks {
-         postBuild = \ _ _ _ _ -> runTestScript
-       , runTests = \ _ _ _ _ -> runTestScript
+       runTests = \ _ _ _ _ -> runTestScript
        }

 runTestScript =
