Index: haskell-debian-3.40/Debian/Relation/Common.hs
===================================================================
--- haskell-debian-3.40.orig/Debian/Relation/Common.hs	2010-05-03 14:39:26.000000000 +0200
+++ haskell-debian-3.40/Debian/Relation/Common.hs	2010-05-03 14:47:46.000000000 +0200
@@ -4,6 +4,7 @@

 import Data.List
 import Text.ParserCombinators.Parsec
+import Data.Function

 -- Local Modules

@@ -63,16 +64,15 @@
     show (GRE v) = " (>= " ++ show v ++ ")"
     show (SGR v) = " (>> " ++ show v ++ ")"

--- |@FIXME:@ This instance is currently incomplete and only handles the case
--- where two version requirements are equal.
+-- |The sort order is based on version number first, then on the kind of
+-- relation, sorting in the order <<, <= , ==, >= , >>
 instance Ord VersionReq where
-    compare r1 r2 =
-	if r1 == r2
-	   then EQ
-	   else
-	case (r1, r2) of
-	     (EEQ v1, EEQ v2) -> compare v1 v2
-	     (a,b) -> error $ "Ord VersionReq does not handle (" ++ show a ++", "++ show b++")"
+    compare = compare `on` extr
+      where extr (SLT v) = (v,0)
+            extr (LTE v) = (v,1)
+            extr (EEQ v) = (v,2)
+            extr (GRE v) = (v,3)
+            extr (SGR v) = (v,4)


 -- |Check if a version number satisfies a version requirement.
