Index: taffybar/taffybar.cabal
===================================================================
--- taffybar.orig/taffybar.cabal	2013-07-17 21:35:44.000000000 +0200
+++ taffybar/taffybar.cabal	2013-07-23 12:54:33.000000000 +0200
@@ -22,9 +22,9 @@
   .
     * More robust strut handling for multiple monitors of different sizes (contributed by Morgan Gibson)
   .
-    * New widgets from José A. Romero (network monitor, fs monitor, another CPU monitor)
+    * New widgets from Jose' A. Romero (network monitor, fs monitor, another CPU monitor)
   .
-    * Allow the bar widget to grow vertically (also contributed by José A. Romero)
+    * Allow the bar widget to grow vertically (also contributed by Jose' A. Romero)
   .
   Changes in v0.2.0:
   .
