--- a/crypto-pubkey-openssh.cabal
+++ b/crypto-pubkey-openssh.cabal
@@ -59,8 +59,8 @@
                   , asn1-types                 == 0.2.*
                   , asn1-encoding              == 0.8.*

-                  , tasty                      == 0.8.*
-                  , tasty-quickcheck           == 0.8.*
+                  , tasty
+                  , tasty-quickcheck
                   , QuickCheck                 >= 2.6
                   , temporary                  == 1.2.*  || == 1.1.*
                   , process
