--- a/crypto-pubkey-openssh.cabal
+++ b/crypto-pubkey-openssh.cabal
@@ -63,7 +63,7 @@
                   , tasty-quickcheck           == 0.8.*
                   , QuickCheck                 >= 2.6
                   , temporary                  == 1.2.*  || == 1.1.*
-                  , process                    == 1.2.*
+                  , process
                   , filepath                   == 1.3.*
                   , deepseq                    == 1.3.*
