Index: haskell-gio-0.12.5.3/gio.cabal
===================================================================
--- haskell-gio-0.12.5.3.orig/gio.cabal	2014-03-01 09:27:27.000000000 +0000
+++ haskell-gio-0.12.5.3/gio.cabal	2014-09-18 17:55:28.540009896 +0000
@@ -83,3 +83,5 @@
         x-Signals-Types: marshal.list
         x-c2hs-Header:  gio/gio.h
         pkgconfig-depends: gio-2.0 >= 2.16.4
+
+        cpp-options: -Ubool
