Index: haskell-presburger/src/Data/Integer/SAT.hs
===================================================================
--- haskell-presburger.orig/src/Data/Integer/SAT.hs
+++ haskell-presburger/src/Data/Integer/SAT.hs
@@ -43,6 +43,7 @@ module Data.Integer.SAT
 
 import Debug.Trace
 
+import           Prelude hiding ((<>))
 import           Data.Map (Map)
 import qualified Data.Map as Map
 import           Data.List(partition)
