Index: haskell-enumerator/lib/Data/Enumerator/Internal.hs
===================================================================
--- haskell-enumerator.orig/lib/Data/Enumerator/Internal.hs
+++ haskell-enumerator/lib/Data/Enumerator/Internal.hs
@@ -71,6 +71,9 @@ instance Monad Stream where
 	Chunks xs >>= f = mconcat (fmap f xs)
 	EOF >>= _ = EOF
 
+instance Semigroup (Stream a) where
+	(<>) = mappend
+
 instance Monoid (Stream a) where
 	mempty = Chunks mempty
 	mappend (Chunks xs) (Chunks ys) = Chunks (xs ++ ys)
