From: Apollon Oikonomopoulos <apollon@skroutz.gr>
Date: Thu, 14 Mar 2013 17:29:36 +0200
Subject: Fix typo in src/Parsers.hs (fix lintian warning)

---
 src/Parsers.hs |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Parsers.hs b/src/Parsers.hs
index 919ce68..26a817b 100644
--- a/src/Parsers.hs
+++ b/src/Parsers.hs
@@ -228,5 +228,5 @@ parseConfig = runParser parseConf fields "Config" . stripComments
 commandsErr :: String
 commandsErr = "commands: this usually means that a command could not" ++
               "\nbe parsed." ++
-              "\nThe error could be located at the begining of the command" ++
+              "\nThe error could be located at the beginning of the command" ++
               "\nwhich follows the offending one."
