diff -Nur -x '*.orig' -x '*~' darcs-2.0.0/src/Darcs/Commands/Init.lhs darcs-2.0.0.new/src/Darcs/Commands/Init.lhs
--- darcs-2.0.0/src/Darcs/Commands/Init.lhs	2008-04-08 01:08:19.000000000 +1000
+++ darcs-2.0.0.new/src/Darcs/Commands/Init.lhs	2008-06-04 13:43:34.000000000 +1000
@@ -56,7 +56,7 @@
 initialize_help =
  "Call 'initialize' once for each project you work on. Run it from the top  \n"++
  "level directory of the project, with the project files already there.   \n"++
- "'initialize' will set up all the directories and files darcs needs in order to \n"++
+ " 'initialize' will set up all the directories and files darcs needs in order to \n"++
  " start keeping track of revisions for your project.  \n"
 \end{code}
