--- a/lib/Filesystem.hs
+++ b/lib/Filesystem.hs
@@ -73,12 +73,6 @@
	, rename
	) where

-#ifndef CABAL_OS_WINDOWS
-#if MIN_VERSION_base(4,2,0)
-#define SYSTEMFILEIO_LOCAL_OPEN_FILE
-#endif
-#endif
-
 import           Prelude hiding (FilePath, readFile, writeFile, appendFile)

 import qualified Control.Exception as Exc
--- a/system-fileio.cabal
+++ b/system-fileio.cabal
@@ -65,6 +65,8 @@
     build-depends:
         unix >= 2.3 && < 2.6
     c-sources: lib/hssystemfileio-unix.c
+    if impl(ghc >= 7.2.0) && impl(ghc < 7.4.0)
+      cpp-options: -DSYSTEMFILEIO_LOCAL_OPEN_FILE

   exposed-modules:
     Filesystem
--- a/tests/system-fileio-tests.cabal
+++ b/tests/system-fileio-tests.cabal
@@ -40,3 +40,5 @@
     build-depends:
         unix >= 2.3 && < 2.6
     c-sources: ../lib/hssystemfileio-unix.c
+    if impl(ghc >= 7.2.0) && impl(ghc < 7.4.0)
+      cpp-options: -DSYSTEMFILEIO_LOCAL_OPEN_FILE
