Index: trifecta-1.5.1.3/tests/doctests.hs
===================================================================
--- trifecta-1.5.1.3.orig/tests/doctests.hs	2015-06-15 14:18:09.316264169 +0200
+++ trifecta-1.5.1.3/tests/doctests.hs	2015-06-15 14:18:09.312264088 +0200
@@ -11,9 +11,9 @@
 main :: IO ()
 main = getSources >>= \sources -> doctest $
     "-isrc"
-  : "-idist/build/autogen"
+  : "-idist-ghc/build/autogen"
   : "-optP-include"
-  : "-optPdist/build/autogen/cabal_macros.h"
+  : "-optPdist-ghc/build/autogen/cabal_macros.h"
   : "-hide-all-packages"
   : map ("-package="++) deps ++ sources
