--- a/DAV.cabal
+++ b/DAV.cabal
@@ -40,7 +40,6 @@ library
                      , http-types >= 0.7
                      , lens >= 3.0
                      , mtl >= 2.1
-                     , mtl-compat
                      , transformers >= 0.3
                      , transformers-base
                      , transformers-compat >= 0.3
@@ -62,7 +61,6 @@ executable hdav
                      , http-types >= 0.7
                      , lens >= 3.0
                      , mtl >= 2.1
-                     , mtl-compat
                      , optparse-applicative >= 0.10.0
                      , transformers >= 0.3
                      , transformers-base
