Index: b/store.cabal
===================================================================
--- a/store.cabal
+++ b/store.cabal
@@ -125,7 +125,7 @@ test-suite store-test
       System.IO.ByteBufferSpec
   hs-source-dirs:
       test
-  ghc-options: -Wall -fwarn-tabs -fwarn-incomplete-uni-patterns -fwarn-incomplete-record-updates -O2 -threaded -rtsopts -with-rtsopts=-N
+  ghc-options: -Wall -fwarn-tabs -fwarn-incomplete-uni-patterns -fwarn-incomplete-record-updates -O2 -threaded -rtsopts
   build-tool-depends:
       hspec-discover:hspec-discover
   build-depends:
@@ -192,7 +192,7 @@ benchmark store-bench
       Paths_store
   hs-source-dirs:
       bench
-  ghc-options: -Wall -fwarn-tabs -fwarn-incomplete-uni-patterns -fwarn-incomplete-record-updates -O2 -threaded -rtsopts -with-rtsopts=-N1 -with-rtsopts=-s -with-rtsopts=-qg
+  ghc-options: -Wall -fwarn-tabs -fwarn-incomplete-uni-patterns -fwarn-incomplete-record-updates -O2 -threaded -rtsopts -with-rtsopts=-s -with-rtsopts=-qg
   build-depends:
       array >=0.5.0.0
     , async >=2.0.2
@@ -264,7 +264,7 @@ benchmark store-weigh
   main-is: Allocations.hs
   hs-source-dirs:
       test
-  ghc-options: -Wall -fwarn-tabs -fwarn-incomplete-uni-patterns -fwarn-incomplete-record-updates -O2 -threaded -rtsopts -with-rtsopts=-N -with-rtsopts=-T -O2
+  ghc-options: -Wall -fwarn-tabs -fwarn-incomplete-uni-patterns -fwarn-incomplete-record-updates -O2 -threaded -rtsopts -with-rtsopts=-T -O2
   build-depends:
       array >=0.5.0.0
     , async >=2.0.2
