--- a/store.cabal
+++ b/store.cabal
@@ -99,7 +99,7 @@
   main-is: Spec.hs
   hs-source-dirs:
       test
-  ghc-options: -Wall -fwarn-tabs -fwarn-incomplete-uni-patterns -fwarn-incomplete-record-updates -O2 -threaded -rtsopts -with-rtsopts=-N
+  ghc-options: -Wall -fwarn-tabs -fwarn-incomplete-uni-patterns -fwarn-incomplete-record-updates -O2 -threaded -rtsopts
   build-depends:
       base >=4.7 && <5
     , store-core >=0.4 && <0.5
@@ -157,7 +157,7 @@
   main-is: Allocations.hs
   hs-source-dirs:
       test
-  ghc-options: -Wall -fwarn-tabs -fwarn-incomplete-uni-patterns -fwarn-incomplete-record-updates -O2 -threaded -rtsopts -with-rtsopts=-N -with-rtsopts=-T -O2
+  ghc-options: -Wall -fwarn-tabs -fwarn-incomplete-uni-patterns -fwarn-incomplete-record-updates -O2 -threaded -rtsopts -with-rtsopts=-T -O2
   build-depends:
       base >=4.7 && <5
     , store-core >=0.4 && <0.5
@@ -215,7 +215,7 @@
   main-is: Bench.hs
   hs-source-dirs:
       bench
-  ghc-options: -Wall -fwarn-tabs -fwarn-incomplete-uni-patterns -fwarn-incomplete-record-updates -O2 -threaded -rtsopts -with-rtsopts=-N1 -with-rtsopts=-s -with-rtsopts=-qg
+  ghc-options: -Wall -fwarn-tabs -fwarn-incomplete-uni-patterns -fwarn-incomplete-record-updates -O2 -threaded -rtsopts -with-rtsopts=-s -with-rtsopts=-qg
   build-depends:
       base >=4.7 && <5
     , store-core >=0.4 && <0.5
