--- a/hledger-lib.cabal
+++ b/hledger-lib.cabal
@@ -70,7 +70,6 @@ library
     , filepath
     , megaparsec >=5.0 && < 5.2
     , mtl
-    , mtl-compat
     , old-time
     , pretty-show >=1.6.4
     , regex-tdfa
@@ -160,7 +159,6 @@ test-suite doctests
     , filepath
     , megaparsec >=5.0 && < 5.2
     , mtl
-    , mtl-compat
     , old-time
     , pretty-show >=1.6.4
     , regex-tdfa
@@ -200,7 +198,6 @@ test-suite hunittests
     , filepath
     , megaparsec >=5.0 && < 5.2
     , mtl
-    , mtl-compat
     , old-time
     , pretty-show >=1.6.4
     , regex-tdfa
