--- a/hledger-lib.cabal
+++ b/hledger-lib.cabal
@@ -73,7 +73,6 @@ library
     , hashtables >= 1.2
     , megaparsec >=5.0 && < 5.3
     , mtl
-    , mtl-compat
     , old-time
     , pretty-show >=1.6.4
     , regex-tdfa
@@ -167,7 +166,6 @@ test-suite doctests
     , hashtables >= 1.2
     , megaparsec >=5.0 && < 5.3
     , mtl
-    , mtl-compat
     , old-time
     , pretty-show >=1.6.4
     , regex-tdfa
@@ -252,7 +250,6 @@ test-suite hunittests
     , hashtables >= 1.2
     , megaparsec >=5.0 && < 5.3
     , mtl
-    , mtl-compat
     , old-time
     , pretty-show >=1.6.4
     , regex-tdfa
