--- a/hledger-lib.cabal
+++ b/hledger-lib.cabal
@@ -120,7 +120,6 @@
     , hashtables >=1.2
     , megaparsec >=5.0
     , mtl
-    , mtl-compat
     , old-time
     , parsec >=3
     , pretty-show >=1.6.4
@@ -212,7 +211,6 @@
     , hashtables >=1.2
     , megaparsec >=5.0
     , mtl
-    , mtl-compat
     , old-time
     , parsec >=3
     , pretty-show >=1.6.4
@@ -303,7 +301,6 @@
     , hledger-lib
     , megaparsec >=5.0
     , mtl
-    , mtl-compat
     , old-time
     , parsec >=3
     , pretty-show >=1.6.4
