--- a/hledger-lib.cabal
+++ b/hledger-lib.cabal
@@ -74,7 +74,6 @@
     , hashtables >= 1.2
     , megaparsec >=5.0 && < 5.2
     , mtl
-    , mtl-compat
     , old-time
     , pretty-show >=1.6.4
     , regex-tdfa
@@ -172,7 +171,6 @@
     , hashtables >= 1.2
     , megaparsec >=5.0 && < 5.2
     , mtl
-    , mtl-compat
     , old-time
     , pretty-show >=1.6.4
     , regex-tdfa
@@ -261,7 +259,6 @@
     , hashtables >= 1.2
     , megaparsec >=5.0 && < 5.2
     , mtl
-    , mtl-compat
     , old-time
     , pretty-show >=1.6.4
     , regex-tdfa
