--- a/hledger-lib.cabal
+++ b/hledger-lib.cabal
@@ -125,7 +125,6 @@
     , hashtables >=1.2.3.1
     , megaparsec >=7.0.0 && <8
     , mtl
-    , mtl-compat
     , old-time
     , parsec >=3
     , parser-combinators >=0.4.0
@@ -225,7 +224,6 @@
     , hashtables >=1.2.3.1
     , megaparsec >=7.0.0 && <8
     , mtl
-    , mtl-compat
     , old-time
     , parsec >=3
     , parser-combinators >=0.4.0
@@ -325,7 +323,6 @@
     , hledger-lib
     , megaparsec >=7.0.0 && <8
     , mtl
-    , mtl-compat
     , old-time
     , parsec >=3
     , parser-combinators >=0.4.0
