--- a/double-conversion.cabal
+++ b/double-conversion.cabal
@@ -62,16 +62,8 @@ flag developer
 library
   c-sources:
     cbits/hs-double-conversion.cc
-    double-conversion/src/bignum.cc
-    double-conversion/src/bignum-dtoa.cc
-    double-conversion/src/cached-powers.cc
-    double-conversion/src/diy-fp.cc
-    double-conversion/src/double-conversion.cc
-    double-conversion/src/fast-dtoa.cc
-    double-conversion/src/fixed-dtoa.cc
-    double-conversion/src/strtod.cc

-  extra-libraries: stdc++
+  extra-libraries: stdc++ double-conversion

   include-dirs:
     double-conversion/src
