--- haskell-binary-orphans-0.1.6.0.orig/binary-orphans.cabal
+++ haskell-binary-orphans-0.1.6.0/binary-orphans.cabal
@@ -41,7 +41,7 @@ library
     , vector-binary-instances  >=0.2.1.0 && <0.3
   if !impl(ghc >= 8.0)
     build-depends:
-        semigroups               >=0.16.2.2  && <0.18.3
+        semigroups               >=0.16.2.2  && <0.18.4
   if !impl(ghc >= 7.10)
     build-depends:
         void                     >=0.7       && <0.8
