--- a/src/Foreign/C/Error/Safe.hs
+++ b/src/Foreign/C/Error/Safe.hs
@@ -99,14 +99,14 @@

 ----------------------------------------------------------------

-eitherErrnoIfMinus1 :: (Num a) => IO a -> IO (Either C.Errno a)
+eitherErrnoIfMinus1 :: (Num a, Eq a) => IO a -> IO (Either C.Errno a)
 eitherErrnoIfMinus1 = eitherErrnoIf (-1 ==)

-eitherErrnoIfMinus1Retry :: (Num a) => IO a -> IO (Either C.Errno a)
+eitherErrnoIfMinus1Retry :: (Num a, Eq a) => IO a -> IO (Either C.Errno a)
 eitherErrnoIfMinus1Retry = eitherErrnoIfRetry (-1 ==)

 eitherErrnoIfMinus1RetryMayBlock
-    :: (Num a) => IO a -> IO b -> IO (Either C.Errno a)
+    :: (Num a, Eq a) => IO a -> IO b -> IO (Either C.Errno a)
 eitherErrnoIfMinus1RetryMayBlock =
     eitherErrnoIfRetryMayBlock (-1 ==)
