Index: haskell-hs-bibutils/src/Text/Bibutils.hsc
===================================================================
--- haskell-hs-bibutils.orig/src/Text/Bibutils.hsc
+++ haskell-hs-bibutils/src/Text/Bibutils.hsc
@@ -93,7 +93,6 @@ module Text.Bibutils
     , isi_out
     , word2007_out
     , adsab_out
-    , nbib_out
 
     -- * Options for Specific Output Formats
     , FormatOpt
@@ -426,7 +425,6 @@ newtype BiblioOut = BiblioOut { unBiblio
  , isi_out      = BIBL_ISIOUT
  , word2007_out = BIBL_WORD2007OUT
  , adsab_out    = BIBL_ADSABSOUT
- , nbib_out     = BIBL_NBIBOUT
  }
 
 newtype FormatOpt = FormatOpt { unFormatOpt :: CInt }
