--- a/process-conduit.cabal
+++ b/process-conduit.cabal
@@ -48,4 +48,6 @@
                      , bytestring
                      , hspec           >= 1.3
                      , conduit
+                     , conduit-extra
                      , process-conduit
+                     , resourcet
--- a/test.hs
+++ b/test.hs
@@ -3,6 +3,7 @@
 import Data.Conduit.Process
 import System.Process.QQ

+import Control.Monad.Trans.Resource (runResourceT)
 import qualified Data.ByteString.Lazy.Char8 as L
 import Data.Conduit
 import qualified Data.Conduit.Binary as CB
