Index: process-conduit-0.5.0.4/process-conduit.cabal
===================================================================
--- process-conduit-0.5.0.4.orig/process-conduit.cabal	2012-11-20 12:59:52.000000000 +0000
+++ process-conduit-0.5.0.4/process-conduit.cabal	2012-11-28 22:44:11.445928771 +0000
@@ -40,7 +40,7 @@

 test-suite process-conduit-test
   type:                exitcode-stdio-1.0
-  hs-source-dirs:      dist
+  hs-source-dirs:      dist-ghc
   main-is:             ../test.hs
   build-depends:       base            == 4.*
                      , bytestring
