Index: haskell-texmath-0.3.0.3/texmath.cabal
===================================================================
--- haskell-texmath-0.3.0.3.orig/texmath.cabal	2010-10-19 20:00:42.000000000 +0200
+++ haskell-texmath-0.3.0.3/texmath.cabal	2010-10-19 20:00:47.000000000 +0200
@@ -56,7 +56,7 @@
   default:           False

 Library
-    Build-depends:       xml, parsec >= 2, containers
+    Build-depends:       xml, parsec >= 3, containers
     if impl(ghc >= 6.10)
       Build-depends: base >= 4 && < 5, syb
     else
