--- a/happstack-authenticate.cabal
+++ b/happstack-authenticate.cabal
@@ -39,7 +39,7 @@
                        containers                   >= 0.4 && < 0.6,
                        ixset                        >= 1.0 && < 1.1,
                        happstack-server             >= 6.0 && < 7.4,
-                       http-conduit                 >= 1.4 && < 2.1,
+                       http-conduit                 >= 1.4 && < 2.2,
                        http-types                   >= 0.6 && < 0.9,
                        fb                           >= 0.13 && < 1.1,
                        safecopy                     >= 0.6,
