Index: b/happstack-authenticate.cabal
===================================================================
--- a/happstack-authenticate.cabal
+++ b/happstack-authenticate.cabal
@@ -68,7 +68,7 @@ Library
                        mime-mail                    >= 0.4  && < 0.6,
                        mtl                          >= 2.0  && < 2.3,
                        lens                         >= 4.2  && < 5.2,
-                       pwstore-purehaskell          == 2.1.*,
+                       pwstore-fast,
                        stm                          >= 2.4  && < 2.6,
                        text                         >= 0.11 && < 2.1,
                        time                         >= 1.2  && < 1.14,
