--- a/happstack-authenticate.cabal
+++ b/happstack-authenticate.cabal
@@ -67,7 +67,7 @@
                        mime-mail                    >= 0.4  && < 0.5,
                        mtl                          >= 2.0  && < 2.3,
                        lens                         >= 4.2  && < 4.15,
-                       pwstore-purehaskell          == 2.1.*,
+                       pwstore-fast,
                        text                         >= 0.11 && < 1.3,
                        time                         >= 1.2  && < 1.7,
                        userid                       >= 0.1  && < 0.2,
@@ -78,4 +78,4 @@
                        web-routes-boomerang         >= 0.28 && < 0.29,
                        web-routes-happstack         == 0.23.*,
                        web-routes-th                >= 0.22 && < 0.23,
-                       web-routes-hsp               >= 0.24 && < 0.25
\ No newline at end of file
+                       web-routes-hsp               >= 0.24 && < 0.25
