Index: haskell-leksah-server-0.10.0.4/leksah-server.cabal
===================================================================
--- haskell-leksah-server-0.10.0.4.orig/leksah-server.cabal	2011-05-27 19:55:00.000000000 +0200
+++ haskell-leksah-server-0.10.0.4/leksah-server.cabal	2011-05-27 19:58:44.000000000 +0200
@@ -36,7 +36,7 @@
                containers >=0.2.0.0 && <0.5,
                directory >=1.0.0.2 && <1.2, filepath >=1.1.0.1 && <1.3, ghc >=6.10.1 && <7.1,
                ltk >=0.9 && <1.0, mtl >=1.1.0.2 && <2.1, parsec >=2.1.0.1 && <3.2,
-               pretty >=1.0.1.0 && <1.1, process-leksah >=1.0.1.3 && <1.1, time >=1.1 && <1.3, deepseq >=1.1 && <1.2,
+               pretty >=1.0.1.0 && <1.1, time >=1.1 && <1.3, deepseq >=1.1 && <1.2,
                hslogger >= 1.0.7 && <1.2, network >=2.2 && <3.0
     if (impl(ghc >= 6.12))
        build-depends: haddock >= 2.7.2 && <2.10
@@ -72,13 +72,27 @@
        ghc-options: -Wall -ferror-spans -O2
     ghc-prof-options: -auto-all -prof

+    exposed-modules:
+        IDE.System.Process
+    if impl(ghc)
+        exposed-modules:
+          IDE.System.Process.Internals
+    c-sources:
+        process-leksah/cbits/runProcess.c
+    include-dirs: process-leksah/include
+    includes:
+        runProcess.h
+    cpp-options: -Dbase4
+    hs-source-dirs: process-leksah
+
+
 executable leksah-server
     build-depends: Cabal >=1.6.0.1 && <1.11, base >= 4.0.0.0 && <4.4, binary >=0.5.0.0 && <0.6,
                binary-shared >=0.8 && <0.9, bytestring >=0.9.0.1 && <0.10,
                containers >=0.2.0.0 && <0.5,
                directory >=1.0.0.2 && <1.2, filepath >=1.1.0.1 && <1.3, ghc >=6.10.1 && <7.1,
                ltk >=0.9 && <1.0, mtl >=1.1.0.2 && <2.1, parsec >=2.1.0.1 && <3.2,
-               pretty >=1.0.1.0 && <1.1, process-leksah >=1.0.1.3 && <1.1, time >=1.1 && <1.3, deepseq >=1.1 && <1.2,
+               pretty >=1.0.1.0 && <1.1, time >=1.1 && <1.3, deepseq >=1.1 && <1.2,
                hslogger >= 1.0.7 && <1.2, network >=2.2 && <3.0
     if (impl(ghc >= 6.12))
        build-depends: haddock >= 2.7.2 && <2.10
@@ -120,6 +134,13 @@
         ghc-options: -Wall -ferror-spans -O2
     ghc-prof-options: -auto-all -prof

+    -- Bundled leksah-process
+    c-sources:
+        process-leksah/cbits/runProcess.c
+    include-dirs: process-leksah/include
+    cpp-options: -Dbase4
+    hs-source-dirs: process-leksah
+
 executable leksahecho
     main-is: LeksahEcho.hs
     buildable: True
@@ -127,8 +148,7 @@
     hs-source-dirs: src
     ghc-prof-options: -auto-all -prof
 --    ghc-shared-options: -auto-all -prof
-    build-depends:  base >= 4.0.0.0 && <4.4, hslogger >= 1.0.7 && <1.2, deepseq >=1.1 && <1.2,
-                    process-leksah >=1.0.1.3 && <1.1
+    build-depends:  base >= 4.0.0.0 && <4.4, hslogger >= 1.0.7 && <1.2, deepseq >=1.1 && <1.2


     if flag(threaded)
@@ -139,4 +159,10 @@
     else
        ghc-options: -Wall -ferror-spans -O2

-
+    -- Bundled leksah-process
+    build-depends: unix
+    c-sources:
+        process-leksah/cbits/runProcess.c
+    include-dirs: process-leksah/include
+    cpp-options: -Dbase4
+    hs-source-dirs: process-leksah
Index: haskell-leksah-server-0.10.0.4/src/IDE/Utils/Tool.hs
===================================================================
--- haskell-leksah-server-0.10.0.4.orig/src/IDE/Utils/Tool.hs	2011-05-27 19:55:43.000000000 +0200
+++ haskell-leksah-server-0.10.0.4/src/IDE/Utils/Tool.hs	2011-05-27 19:55:52.000000000 +0200
@@ -41,15 +41,9 @@
 import Control.Monad (unless, when)
 import Data.List (stripPrefix)
 import Data.Maybe (isJust, catMaybes)
-#ifdef MIN_VERSION_process_leksah
 import IDE.System.Process
        (proc, waitForProcess, ProcessHandle, createProcess, CreateProcess(..))
 import IDE.System.Process.Internals (StdStream(..))
-#else
-import System.Process
-       (proc, waitForProcess, ProcessHandle, createProcess, CreateProcess(..))
-import System.Process.Internals (StdStream(..))
-#endif
 import Control.DeepSeq
 import System.Log.Logger (debugM, criticalM)
 import System.Exit (ExitCode(..))
Index: haskell-leksah-server-0.10.0.4/src/IDE/Utils/FileUtils.hs
===================================================================
--- haskell-leksah-server-0.10.0.4.orig/src/IDE/Utils/FileUtils.hs	2011-05-27 19:56:08.000000000 +0200
+++ haskell-leksah-server-0.10.0.4/src/IDE/Utils/FileUtils.hs	2011-05-27 19:56:19.000000000 +0200
@@ -71,13 +71,8 @@
 import qualified Data.Set as  Set (empty, fromList)
 import Control.Monad.Trans (MonadIO(..))
 import Distribution.Package (PackageIdentifier)
-#ifdef MIN_VERSION_process_leksah
 import IDE.System.Process
     (waitForProcess, runCommand)
-#else
-import System.Process
-    (waitForProcess, runCommand)
-#endif
 import Data.Char (ord)
 import Distribution.Text (simpleParse)

Index: haskell-leksah-server-0.10.0.4/src/IDE/Metainfo/PackageCollector.hs
===================================================================
--- haskell-leksah-server-0.10.0.4.orig/src/IDE/Metainfo/PackageCollector.hs	2011-05-27 19:56:35.000000000 +0200
+++ haskell-leksah-server-0.10.0.4/src/IDE/Metainfo/PackageCollector.hs	2011-05-27 19:56:48.000000000 +0200
@@ -49,11 +49,7 @@
 import Control.Monad (when)
 import System.IO (withBinaryFile, IOMode(..), hPutStr)
 #else
-#ifdef MIN_VERSION_process_leksah
 import IDE.System.Process (system)
-#else
-import System.Process (system)
-#endif
 #endif
 import Prelude hiding(catch)
