Index: hamlet-1.1.1/hamlet.cabal
===================================================================
--- hamlet-1.1.1.orig/hamlet.cabal	2012-09-24 14:14:22.000000000 +0000
+++ hamlet-1.1.1/hamlet.cabal	2012-10-20 16:54:08.564870519 +0000
@@ -69,7 +69,7 @@
     build-depends: hamlet
                  , base             >= 4       && < 5
                  , parsec           >= 2       && < 4
-                 , containers       >= 0.2     && < 0.5
+                 , containers       >= 0.2     && < 0.6
                  , text             >= 0.7     && < 1
                  , HUnit
                  , hspec            >= 1.3
