--- a/reform-hamlet.cabal
+++ b/reform-hamlet.cabal
@@ -22,7 +22,7 @@
                        Text.Reform.Hamlet.String
                        Text.Reform.Hamlet.Text
   Build-depends:       base         > 4     && < 5,
-                       blaze-markup > 0.5   && < 0.7,
+                       blaze-markup > 0.5   && < 0.8,
                        shakespeare  >= 2.0  && < 2.1,
                        reform       == 0.2.*,
                        text         >= 0.11 && < 1.3
